// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/DeleteDocumentsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDeleteDocumentsRequest = instanceOfDeleteDocumentsRequest;
exports.DeleteDocumentsRequestFromJSON = DeleteDocumentsRequestFromJSON;
exports.DeleteDocumentsRequestFromJSONTyped = DeleteDocumentsRequestFromJSONTyped;
exports.DeleteDocumentsRequestToJSON = DeleteDocumentsRequestToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the DeleteDocumentsRequest interface.
 */
function instanceOfDeleteDocumentsRequest(value) {
    let isInstance = true;
    return isInstance;
}
function DeleteDocumentsRequestFromJSON(json) {
    return DeleteDocumentsRequestFromJSONTyped(json, false);
}
function DeleteDocumentsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'ids': !(0, runtime_1.exists)(json, 'ids') ? undefined : json['ids'],
        'deleteAll': !(0, runtime_1.exists)(json, 'delete_all') ? undefined : json['delete_all'],
    };
}
function DeleteDocumentsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'ids': value.ids,
        'delete_all': value.deleteAll,
    };
}
