// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/DocumentSearchMatch.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDocumentSearchMatch = instanceOfDocumentSearchMatch;
exports.DocumentSearchMatchFromJSON = DocumentSearchMatchFromJSON;
exports.DocumentSearchMatchFromJSONTyped = DocumentSearchMatchFromJSONTyped;
exports.DocumentSearchMatchToJSON = DocumentSearchMatchToJSON;
/**
 * Check if a given object implements the DocumentSearchMatch interface.
 */
function instanceOfDocumentSearchMatch(value) {
    let isInstance = true;
    isInstance = isInstance && "_id" in value;
    isInstance = isInstance && "_score" in value;
    return isInstance;
}
function DocumentSearchMatchFromJSON(json) {
    return DocumentSearchMatchFromJSONTyped(json, false);
}
function DocumentSearchMatchFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        ...json,
        '_id': json['_id'],
        '_score': json['_score'],
    };
}
function DocumentSearchMatchToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        ...value,
        '_id': value._id,
        '_score': value._score,
    };
}
