// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/DocumentScoringMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDocumentScoringMethod = instanceOfDocumentScoringMethod;
exports.DocumentScoringMethodFromJSON = DocumentScoringMethodFromJSON;
exports.DocumentScoringMethodFromJSONTyped = DocumentScoringMethodFromJSONTyped;
exports.DocumentScoringMethodToJSON = DocumentScoringMethodToJSON;
const runtime_1 = require("../runtime");
const SparseValues_1 = require("./SparseValues");
/**
 * Check if a given object implements the DocumentScoringMethod interface.
 */
function instanceOfDocumentScoringMethod(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function DocumentScoringMethodFromJSON(json) {
    return DocumentScoringMethodFromJSONTyped(json, false);
}
function DocumentScoringMethodFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'field': !(0, runtime_1.exists)(json, 'field') ? undefined : json['field'],
        'query': !(0, runtime_1.exists)(json, 'query') ? undefined : json['query'],
        'values': !(0, runtime_1.exists)(json, 'values') ? undefined : json['values'],
        'sparseValues': !(0, runtime_1.exists)(json, 'sparse_values') ? undefined : (0, SparseValues_1.SparseValuesFromJSON)(json['sparse_values']),
    };
}
function DocumentScoringMethodToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'field': value.field,
        'query': value.query,
        'values': value.values,
        'sparse_values': (0, SparseValues_1.SparseValuesToJSON)(value.sparseValues),
    };
}
