// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/FetchDocumentsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFetchDocumentsResponse = instanceOfFetchDocumentsResponse;
exports.FetchDocumentsResponseFromJSON = FetchDocumentsResponseFromJSON;
exports.FetchDocumentsResponseFromJSONTyped = FetchDocumentsResponseFromJSONTyped;
exports.FetchDocumentsResponseToJSON = FetchDocumentsResponseToJSON;
const runtime_1 = require("../runtime");
const DocumentFetchUsage_1 = require("./DocumentFetchUsage");
const FetchedDocumentRecord_1 = require("./FetchedDocumentRecord");
/**
 * Check if a given object implements the FetchDocumentsResponse interface.
 */
function instanceOfFetchDocumentsResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "documents" in value;
    isInstance = isInstance && "namespace" in value;
    isInstance = isInstance && "usage" in value;
    return isInstance;
}
function FetchDocumentsResponseFromJSON(json) {
    return FetchDocumentsResponseFromJSONTyped(json, false);
}
function FetchDocumentsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'documents': ((0, runtime_1.mapValues)(json['documents'], FetchedDocumentRecord_1.FetchedDocumentRecordFromJSON)),
        'namespace': json['namespace'],
        'usage': (0, DocumentFetchUsage_1.DocumentFetchUsageFromJSON)(json['usage']),
    };
}
function FetchDocumentsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'documents': ((0, runtime_1.mapValues)(value.documents, FetchedDocumentRecord_1.FetchedDocumentRecordToJSON)),
        'namespace': value.namespace,
        'usage': (0, DocumentFetchUsage_1.DocumentFetchUsageToJSON)(value.usage),
    };
}
