// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/FetchDocumentsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFetchDocumentsRequest = instanceOfFetchDocumentsRequest;
exports.FetchDocumentsRequestFromJSON = FetchDocumentsRequestFromJSON;
exports.FetchDocumentsRequestFromJSONTyped = FetchDocumentsRequestFromJSONTyped;
exports.FetchDocumentsRequestToJSON = FetchDocumentsRequestToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the FetchDocumentsRequest interface.
 */
function instanceOfFetchDocumentsRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "ids" in value;
    return isInstance;
}
function FetchDocumentsRequestFromJSON(json) {
    return FetchDocumentsRequestFromJSONTyped(json, false);
}
function FetchDocumentsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'ids': json['ids'],
        'includeFields': !(0, runtime_1.exists)(json, 'include_fields') ? undefined : json['include_fields'],
    };
}
function FetchDocumentsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'ids': value.ids,
        'include_fields': value.includeFields,
    };
}
