// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/SearchDocumentsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSearchDocumentsResponse = instanceOfSearchDocumentsResponse;
exports.SearchDocumentsResponseFromJSON = SearchDocumentsResponseFromJSON;
exports.SearchDocumentsResponseFromJSONTyped = SearchDocumentsResponseFromJSONTyped;
exports.SearchDocumentsResponseToJSON = SearchDocumentsResponseToJSON;
const DocumentSearchMatch_1 = require("./DocumentSearchMatch");
const DocumentSearchUsage_1 = require("./DocumentSearchUsage");
/**
 * Check if a given object implements the SearchDocumentsResponse interface.
 */
function instanceOfSearchDocumentsResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "matches" in value;
    isInstance = isInstance && "namespace" in value;
    isInstance = isInstance && "usage" in value;
    return isInstance;
}
function SearchDocumentsResponseFromJSON(json) {
    return SearchDocumentsResponseFromJSONTyped(json, false);
}
function SearchDocumentsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'matches': (json['matches'].map(DocumentSearchMatch_1.DocumentSearchMatchFromJSON)),
        'namespace': json['namespace'],
        'usage': (0, DocumentSearchUsage_1.DocumentSearchUsageFromJSON)(json['usage']),
    };
}
function SearchDocumentsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'matches': (value.matches.map(DocumentSearchMatch_1.DocumentSearchMatchToJSON)),
        'namespace': value.namespace,
        'usage': (0, DocumentSearchUsage_1.DocumentSearchUsageToJSON)(value.usage),
    };
}
