// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/SearchDocumentsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfSearchDocumentsRequest = instanceOfSearchDocumentsRequest;
exports.SearchDocumentsRequestFromJSON = SearchDocumentsRequestFromJSON;
exports.SearchDocumentsRequestFromJSONTyped = SearchDocumentsRequestFromJSONTyped;
exports.SearchDocumentsRequestToJSON = SearchDocumentsRequestToJSON;
const runtime_1 = require("../runtime");
const DocumentScoringMethod_1 = require("./DocumentScoringMethod");
/**
 * Check if a given object implements the SearchDocumentsRequest interface.
 */
function instanceOfSearchDocumentsRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "scoreBy" in value;
    isInstance = isInstance && "topK" in value;
    return isInstance;
}
function SearchDocumentsRequestFromJSON(json) {
    return SearchDocumentsRequestFromJSONTyped(json, false);
}
function SearchDocumentsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'scoreBy': (json['score_by'].map(DocumentScoringMethod_1.DocumentScoringMethodFromJSON)),
        'topK': json['top_k'],
        'includeFields': !(0, runtime_1.exists)(json, 'include_fields') ? undefined : json['include_fields'],
        'filter': !(0, runtime_1.exists)(json, 'filter') ? undefined : json['filter'],
    };
}
function SearchDocumentsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'score_by': (value.scoreBy.map(DocumentScoringMethod_1.DocumentScoringMethodToJSON)),
        'top_k': value.topK,
        'include_fields': value.includeFields,
        'filter': value.filter,
    };
}
