// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/UpdateDocumentsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateDocumentsRequest = instanceOfUpdateDocumentsRequest;
exports.UpdateDocumentsRequestFromJSON = UpdateDocumentsRequestFromJSON;
exports.UpdateDocumentsRequestFromJSONTyped = UpdateDocumentsRequestFromJSONTyped;
exports.UpdateDocumentsRequestToJSON = UpdateDocumentsRequestToJSON;
const UpdateDocumentRecord_1 = require("./UpdateDocumentRecord");
/**
 * Check if a given object implements the UpdateDocumentsRequest interface.
 */
function instanceOfUpdateDocumentsRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "documents" in value;
    return isInstance;
}
function UpdateDocumentsRequestFromJSON(json) {
    return UpdateDocumentsRequestFromJSONTyped(json, false);
}
function UpdateDocumentsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'documents': (json['documents'].map(UpdateDocumentRecord_1.UpdateDocumentRecordFromJSON)),
    };
}
function UpdateDocumentsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'documents': (value.documents.map(UpdateDocumentRecord_1.UpdateDocumentRecordToJSON)),
    };
}
