// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/UpsertDocumentsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpsertDocumentsRequest = instanceOfUpsertDocumentsRequest;
exports.UpsertDocumentsRequestFromJSON = UpsertDocumentsRequestFromJSON;
exports.UpsertDocumentsRequestFromJSONTyped = UpsertDocumentsRequestFromJSONTyped;
exports.UpsertDocumentsRequestToJSON = UpsertDocumentsRequestToJSON;
const DocumentRecord_1 = require("./DocumentRecord");
/**
 * Check if a given object implements the UpsertDocumentsRequest interface.
 */
function instanceOfUpsertDocumentsRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "documents" in value;
    return isInstance;
}
function UpsertDocumentsRequestFromJSON(json) {
    return UpsertDocumentsRequestFromJSONTyped(json, false);
}
function UpsertDocumentsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'documents': (json['documents'].map(DocumentRecord_1.DocumentRecordFromJSON)),
    };
}
function UpsertDocumentsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'documents': (value.documents.map(DocumentRecord_1.DocumentRecordToJSON)),
    };
}
