// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/UpsertRecord.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpsertRecord = instanceOfUpsertRecord;
exports.UpsertRecordFromJSON = UpsertRecordFromJSON;
exports.UpsertRecordFromJSONTyped = UpsertRecordFromJSONTyped;
exports.UpsertRecordToJSON = UpsertRecordToJSON;
/**
 * Check if a given object implements the UpsertRecord interface.
 */
function instanceOfUpsertRecord(value) {
    let isInstance = true;
    isInstance = isInstance && "_id" in value;
    return isInstance;
}
function UpsertRecordFromJSON(json) {
    return UpsertRecordFromJSONTyped(json, false);
}
function UpsertRecordFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        '_id': json['_id'],
    };
}
function UpsertRecordToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '_id': value._id,
    };
}
