// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/models/UpsertDocumentsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpsertDocumentsResponse = instanceOfUpsertDocumentsResponse;
exports.UpsertDocumentsResponseFromJSON = UpsertDocumentsResponseFromJSON;
exports.UpsertDocumentsResponseFromJSONTyped = UpsertDocumentsResponseFromJSONTyped;
exports.UpsertDocumentsResponseToJSON = UpsertDocumentsResponseToJSON;
/**
 * Check if a given object implements the UpsertDocumentsResponse interface.
 */
function instanceOfUpsertDocumentsResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "upsertedCount" in value;
    return isInstance;
}
function UpsertDocumentsResponseFromJSON(json) {
    return UpsertDocumentsResponseFromJSONTyped(json, false);
}
function UpsertDocumentsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'upsertedCount': json['upserted_count'],
    };
}
function UpsertDocumentsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'upserted_count': value.upsertedCount,
    };
}
