// @pinecone-database/pinecone/dist/preview/indexes/alphaDataOperationsProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlphaDocumentOperationsProvider = void 0;
const db_data_1 = require("../../pinecone-generated-ts-fetch-alpha/db_data");
const utils_1 = require("../../utils");
const middleware_1 = require("../../utils/middleware");
const indexHostSingleton_1 = require("../../data/indexHostSingleton");
const errors_1 = require("../../errors");
class AlphaDocumentOperationsProvider {
    config;
    indexName;
    indexHostUrl;
    additionalHeaders;
    _api;
    constructor(config, indexName, indexHostUrl, additionalHeaders) {
        this.config = config;
        this.indexName = indexName;
        this.indexHostUrl = (0, utils_1.normalizeUrl)(indexHostUrl);
        this.additionalHeaders = additionalHeaders;
    }
    async provide() {
        if (this._api) {
            return this._api;
        }
        if (!this.indexHostUrl) {
            if (!this.indexName) {
                throw new errors_1.PineconeArgumentError('Either name or host must be provided in IndexOptions');
            }
            this.indexHostUrl = await indexHostSingleton_1.IndexHostSingleton.getHostUrl(this.config, this.indexName);
        }
        this._api = this._buildApi();
        return this._api;
    }
    _buildApi() {
        const headers = this.additionalHeaders || null;
        const apiConfig = {
            basePath: this.indexHostUrl,
            apiKey: this.config.apiKey,
            queryParamsStringify: utils_1.queryParamsStringify,
            headers: {
                'User-Agent': (0, utils_1.buildUserAgent)(this.config),
                'X-Pinecone-Api-Version': db_data_1.X_PINECONE_API_VERSION,
                ...headers,
            },
            fetchApi: (0, utils_1.getFetch)(this.config),
            middleware: (0, middleware_1.createMiddlewareArray)(),
        };
        return new db_data_1.DocumentOperationsApi(new db_data_1.Configuration(apiConfig));
    }
}
exports.AlphaDocumentOperationsProvider = AlphaDocumentOperationsProvider;
