// @pinecone-database/pinecone/dist/preview/indexes/createIndexFromBackup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createIndexFromBackup = void 0;
const db_control_1 = require("../../pinecone-generated-ts-fetch-alpha/db_control");
const errors_1 = require("../../errors");
const handling_1 = require("../../errors/handling");
/**
 * Creates an index from a Pinecone backup using the 2026-01.alpha API.
 *
 * The creation is accepted asynchronously — returns a `restoreJobId` and
 * `indexId` once the request is accepted (HTTP 202). Poll the restore job
 * to track completion.
 *
 * @param api - The alpha manage-indexes API client.
 * @param backupId - The ID of the backup to create an index from.
 * @param options - Configuration for the new index.
 * @see [Backups](https://docs.pinecone.io/guides/indexes/backups)
 * @alpha
 */
const createIndexFromBackup = async (api, backupId, options) => {
    if (!backupId) {
        throw new errors_1.PineconeArgumentError('You must pass a non-empty string for `backupId` to create an index from a backup.');
    }
    if (!options.name) {
        throw new errors_1.PineconeArgumentError('You must pass a non-empty string for `name` to create an index from a backup.');
    }
    try {
        return await api.createIndexFromBackupOperation({
            backupId,
            createIndexFromBackupRequest: {
                name: options.name,
                tags: options.tags,
                deletionProtection: options.deletionProtection,
                readCapacity: options.readCapacity,
            },
            xPineconeApiVersion: db_control_1.X_PINECONE_API_VERSION,
        });
    }
    catch (e) {
        throw await (0, handling_1.handleApiError)(e, async (_, rawMessageText) => `Error creating index from backup ${backupId}: ${rawMessageText}`);
    }
};
exports.createIndexFromBackup = createIndexFromBackup;
