// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_control/models/ListAssistantsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant control plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge  from millions of documents at scale. This API supports creating and managing assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListAssistantsResponse = instanceOfListAssistantsResponse;
exports.ListAssistantsResponseFromJSON = ListAssistantsResponseFromJSON;
exports.ListAssistantsResponseFromJSONTyped = ListAssistantsResponseFromJSONTyped;
exports.ListAssistantsResponseToJSON = ListAssistantsResponseToJSON;
const runtime_1 = require("../runtime");
const Assistant_1 = require("./Assistant");
const PaginationResponse_1 = require("./PaginationResponse");
/**
 * Check if a given object implements the ListAssistantsResponse interface.
 */
function instanceOfListAssistantsResponse(value) {
    let isInstance = true;
    return isInstance;
}
function ListAssistantsResponseFromJSON(json) {
    return ListAssistantsResponseFromJSONTyped(json, false);
}
function ListAssistantsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'assistants': !(0, runtime_1.exists)(json, 'assistants') ? undefined : (json['assistants'].map(Assistant_1.AssistantFromJSON)),
        'pagination': !(0, runtime_1.exists)(json, 'pagination') ? undefined : (0, PaginationResponse_1.PaginationResponseFromJSON)(json['pagination']),
    };
}
function ListAssistantsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'assistants': value.assistants === undefined ? undefined : (value.assistants.map(Assistant_1.AssistantToJSON)),
        'pagination': (0, PaginationResponse_1.PaginationResponseToJSON)(value.pagination),
    };
}
