// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_control/models/UpdateAssistantResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant control plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge  from millions of documents at scale. This API supports creating and managing assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateAssistantResponse = instanceOfUpdateAssistantResponse;
exports.UpdateAssistantResponseFromJSON = UpdateAssistantResponseFromJSON;
exports.UpdateAssistantResponseFromJSONTyped = UpdateAssistantResponseFromJSONTyped;
exports.UpdateAssistantResponseToJSON = UpdateAssistantResponseToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateAssistantResponse interface.
 */
function instanceOfUpdateAssistantResponse(value) {
    let isInstance = true;
    return isInstance;
}
function UpdateAssistantResponseFromJSON(json) {
    return UpdateAssistantResponseFromJSONTyped(json, false);
}
function UpdateAssistantResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'assistantName': !(0, runtime_1.exists)(json, 'assistant_name') ? undefined : json['assistant_name'],
        'instructions': !(0, runtime_1.exists)(json, 'instructions') ? undefined : json['instructions'],
        'metadata': !(0, runtime_1.exists)(json, 'metadata') ? undefined : json['metadata'],
    };
}
function UpdateAssistantResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'assistant_name': value.assistantName,
        'instructions': value.instructions,
        'metadata': value.metadata,
    };
}
