// @pinecone-database/pinecone/dist/integration/preview-data/searchDocuments.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const index_1 = require("../../index");
const test_helpers_1 = require("../test-helpers");
let pc;
let indexName;
const namespace = 'test-ns';
beforeAll(async () => {
    pc = new index_1.Pinecone();
    indexName = (0, test_helpers_1.randomName)('preview-search-docs');
    await pc.preview.indexes.create({
        name: indexName,
        schema: {
            fields: {
                content: { type: 'string', fullTextSearch: {} },
            },
        },
        waitUntilReady: true,
    });
    await pc.preview.index({ name: indexName }).upsertDocuments(namespace, {
        documents: [
            {
                _id: 'doc-1',
                content: 'Machine learning is fascinating',
                title: 'ML Doc',
            },
            {
                _id: 'doc-2',
                content: 'Vector databases enable semantic search',
                title: 'VDB Doc',
            },
        ],
    });
    // Wait for propagation
    await new Promise((resolve) => setTimeout(resolve, 8000));
}, 300000);
afterAll(async () => {
    try {
        await pc.preview.indexes.delete(indexName);
    }
    finally {
        // cleanup attempt completed
    }
});
describe('preview searchDocuments', () => {
    test('returns matches, namespace, and usage from search', async () => {
        const result = await pc.preview
            .index({ name: indexName })
            .searchDocuments(namespace, {
            scoreBy: [
                { type: 'text', field: 'content', query: 'machine learning' },
            ],
            topK: 5,
        });
        expect(Array.isArray(result.matches)).toBe(true);
        expect(result.namespace).toBe(namespace);
        expect(result.usage.readUnits).toBeGreaterThanOrEqual(0);
    });
});
