// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/AssistantFileModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAssistantFileModel = instanceOfAssistantFileModel;
exports.AssistantFileModelFromJSON = AssistantFileModelFromJSON;
exports.AssistantFileModelFromJSONTyped = AssistantFileModelFromJSONTyped;
exports.AssistantFileModelToJSON = AssistantFileModelToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AssistantFileModel interface.
 */
function instanceOfAssistantFileModel(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "id" in value;
    return isInstance;
}
function AssistantFileModelFromJSON(json) {
    return AssistantFileModelFromJSONTyped(json, false);
}
function AssistantFileModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'id': json['id'],
        'size': !(0, runtime_1.exists)(json, 'size') ? undefined : json['size'],
        'metadata': !(0, runtime_1.exists)(json, 'metadata') ? undefined : json['metadata'],
        'createdOn': !(0, runtime_1.exists)(json, 'created_on') ? undefined : (new Date(json['created_on'])),
        'updatedOn': !(0, runtime_1.exists)(json, 'updated_on') ? undefined : (new Date(json['updated_on'])),
        'status': !(0, runtime_1.exists)(json, 'status') ? undefined : json['status'],
        'signedUrl': !(0, runtime_1.exists)(json, 'signed_url') ? undefined : json['signed_url'],
        'multimodal': !(0, runtime_1.exists)(json, 'multimodal') ? undefined : json['multimodal'],
    };
}
function AssistantFileModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'id': value.id,
        'size': value.size,
        'metadata': value.metadata,
        'created_on': value.createdOn === undefined ? undefined : (value.createdOn.toISOString()),
        'updated_on': value.updatedOn === undefined ? undefined : (value.updatedOn.toISOString()),
        'status': value.status,
        'signed_url': value.signedUrl,
        'multimodal': value.multimodal,
    };
}
