// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/CitationChunkModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCitationChunkModel = instanceOfCitationChunkModel;
exports.CitationChunkModelFromJSON = CitationChunkModelFromJSON;
exports.CitationChunkModelFromJSONTyped = CitationChunkModelFromJSONTyped;
exports.CitationChunkModelToJSON = CitationChunkModelToJSON;
const CitationModel_1 = require("./CitationModel");
/**
 * Check if a given object implements the CitationChunkModel interface.
 */
function instanceOfCitationChunkModel(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "citation" in value;
    return isInstance;
}
function CitationChunkModelFromJSON(json) {
    return CitationChunkModelFromJSONTyped(json, false);
}
function CitationChunkModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'id': json['id'],
        'model': json['model'],
        'citation': (0, CitationModel_1.CitationModelFromJSON)(json['citation']),
    };
}
function CitationChunkModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'id': value.id,
        'model': value.model,
        'citation': (0, CitationModel_1.CitationModelToJSON)(value.citation),
    };
}
