// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/ContentChunkModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfContentChunkModel = instanceOfContentChunkModel;
exports.ContentChunkModelFromJSON = ContentChunkModelFromJSON;
exports.ContentChunkModelFromJSONTyped = ContentChunkModelFromJSONTyped;
exports.ContentChunkModelToJSON = ContentChunkModelToJSON;
const runtime_1 = require("../runtime");
const ContentChunkModelDelta_1 = require("./ContentChunkModelDelta");
const ContentFilterResults_1 = require("./ContentFilterResults");
/**
 * Check if a given object implements the ContentChunkModel interface.
 */
function instanceOfContentChunkModel(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "delta" in value;
    return isInstance;
}
function ContentChunkModelFromJSON(json) {
    return ContentChunkModelFromJSONTyped(json, false);
}
function ContentChunkModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'id': json['id'],
        'model': json['model'],
        'delta': (0, ContentChunkModelDelta_1.ContentChunkModelDeltaFromJSON)(json['delta']),
        'contentFilterResults': !(0, runtime_1.exists)(json, 'content_filter_results') ? undefined : (0, ContentFilterResults_1.ContentFilterResultsFromJSON)(json['content_filter_results']),
    };
}
function ContentChunkModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'id': value.id,
        'model': value.model,
        'delta': (0, ContentChunkModelDelta_1.ContentChunkModelDeltaToJSON)(value.delta),
        'content_filter_results': (0, ContentFilterResults_1.ContentFilterResultsToJSON)(value.contentFilterResults),
    };
}
