// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/ContentFilterResults.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfContentFilterResults = instanceOfContentFilterResults;
exports.ContentFilterResultsFromJSON = ContentFilterResultsFromJSON;
exports.ContentFilterResultsFromJSONTyped = ContentFilterResultsFromJSONTyped;
exports.ContentFilterResultsToJSON = ContentFilterResultsToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ContentFilterResults interface.
 */
function instanceOfContentFilterResults(value) {
    let isInstance = true;
    return isInstance;
}
function ContentFilterResultsFromJSON(json) {
    return ContentFilterResultsFromJSONTyped(json, false);
}
function ContentFilterResultsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'spec': !(0, runtime_1.exists)(json, 'spec') ? undefined : json['spec'],
        'results': !(0, runtime_1.exists)(json, 'results') ? undefined : json['results'],
    };
}
function ContentFilterResultsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'spec': value.spec,
        'results': value.results,
    };
}
