// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/ContentChunkModelDelta.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfContentChunkModelDelta = instanceOfContentChunkModelDelta;
exports.ContentChunkModelDeltaFromJSON = ContentChunkModelDeltaFromJSON;
exports.ContentChunkModelDeltaFromJSONTyped = ContentChunkModelDeltaFromJSONTyped;
exports.ContentChunkModelDeltaToJSON = ContentChunkModelDeltaToJSON;
/**
 * Check if a given object implements the ContentChunkModelDelta interface.
 */
function instanceOfContentChunkModelDelta(value) {
    let isInstance = true;
    isInstance = isInstance && "content" in value;
    return isInstance;
}
function ContentChunkModelDeltaFromJSON(json) {
    return ContentChunkModelDeltaFromJSONTyped(json, false);
}
function ContentChunkModelDeltaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'content': json['content'],
    };
}
function ContentChunkModelDeltaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'content': value.content,
    };
}
