// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/ListFilesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfListFilesResponse = instanceOfListFilesResponse;
exports.ListFilesResponseFromJSON = ListFilesResponseFromJSON;
exports.ListFilesResponseFromJSONTyped = ListFilesResponseFromJSONTyped;
exports.ListFilesResponseToJSON = ListFilesResponseToJSON;
const runtime_1 = require("../runtime");
const AssistantFileModel_1 = require("./AssistantFileModel");
const PaginationResponse_1 = require("./PaginationResponse");
/**
 * Check if a given object implements the ListFilesResponse interface.
 */
function instanceOfListFilesResponse(value) {
    let isInstance = true;
    return isInstance;
}
function ListFilesResponseFromJSON(json) {
    return ListFilesResponseFromJSONTyped(json, false);
}
function ListFilesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'files': !(0, runtime_1.exists)(json, 'files') ? undefined : (json['files'].map(AssistantFileModel_1.AssistantFileModelFromJSON)),
        'pagination': !(0, runtime_1.exists)(json, 'pagination') ? undefined : (0, PaginationResponse_1.PaginationResponseFromJSON)(json['pagination']),
    };
}
function ListFilesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'files': value.files === undefined ? undefined : (value.files.map(AssistantFileModel_1.AssistantFileModelToJSON)),
        'pagination': (0, PaginationResponse_1.PaginationResponseToJSON)(value.pagination),
    };
}
