// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/MessageEndModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMessageEndModel = instanceOfMessageEndModel;
exports.MessageEndModelFromJSON = MessageEndModelFromJSON;
exports.MessageEndModelFromJSONTyped = MessageEndModelFromJSONTyped;
exports.MessageEndModelToJSON = MessageEndModelToJSON;
const runtime_1 = require("../runtime");
const ContentFilterResults_1 = require("./ContentFilterResults");
const UsageModel_1 = require("./UsageModel");
/**
 * Check if a given object implements the MessageEndModel interface.
 */
function instanceOfMessageEndModel(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "model" in value;
    isInstance = isInstance && "finishReason" in value;
    return isInstance;
}
function MessageEndModelFromJSON(json) {
    return MessageEndModelFromJSONTyped(json, false);
}
function MessageEndModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'id': json['id'],
        'model': json['model'],
        'finishReason': json['finish_reason'],
        'usage': !(0, runtime_1.exists)(json, 'usage') ? undefined : (0, UsageModel_1.UsageModelFromJSON)(json['usage']),
        'contentFilterResults': !(0, runtime_1.exists)(json, 'content_filter_results') ? undefined : (0, ContentFilterResults_1.ContentFilterResultsFromJSON)(json['content_filter_results']),
    };
}
function MessageEndModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'id': value.id,
        'model': value.model,
        'finish_reason': value.finishReason,
        'usage': (0, UsageModel_1.UsageModelToJSON)(value.usage),
        'content_filter_results': (0, ContentFilterResults_1.ContentFilterResultsToJSON)(value.contentFilterResults),
    };
}
