// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/OperationList.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfOperationList = instanceOfOperationList;
exports.OperationListFromJSON = OperationListFromJSON;
exports.OperationListFromJSONTyped = OperationListFromJSONTyped;
exports.OperationListToJSON = OperationListToJSON;
const runtime_1 = require("../runtime");
const OperationModel_1 = require("./OperationModel");
const PaginationResponse_1 = require("./PaginationResponse");
/**
 * Check if a given object implements the OperationList interface.
 */
function instanceOfOperationList(value) {
    let isInstance = true;
    return isInstance;
}
function OperationListFromJSON(json) {
    return OperationListFromJSONTyped(json, false);
}
function OperationListFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'operations': !(0, runtime_1.exists)(json, 'operations') ? undefined : (json['operations'].map(OperationModel_1.OperationModelFromJSON)),
        'pagination': !(0, runtime_1.exists)(json, 'pagination') ? undefined : (0, PaginationResponse_1.PaginationResponseFromJSON)(json['pagination']),
    };
}
function OperationListToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'operations': value.operations === undefined ? undefined : (value.operations.map(OperationModel_1.OperationModelToJSON)),
        'pagination': (0, PaginationResponse_1.PaginationResponseToJSON)(value.pagination),
    };
}
