// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/OperationModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfOperationModel = instanceOfOperationModel;
exports.OperationModelFromJSON = OperationModelFromJSON;
exports.OperationModelFromJSONTyped = OperationModelFromJSONTyped;
exports.OperationModelToJSON = OperationModelToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the OperationModel interface.
 */
function instanceOfOperationModel(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "operationType" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "createdOn" in value;
    return isInstance;
}
function OperationModelFromJSON(json) {
    return OperationModelFromJSONTyped(json, false);
}
function OperationModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'operationType': json['operation_type'],
        'fileId': !(0, runtime_1.exists)(json, 'file_id') ? undefined : json['file_id'],
        'status': json['status'],
        'createdOn': (new Date(json['created_on'])),
        'completedOn': !(0, runtime_1.exists)(json, 'completed_on') ? undefined : (json['completed_on'] === null ? null : new Date(json['completed_on'])),
        'percentComplete': !(0, runtime_1.exists)(json, 'percent_complete') ? undefined : json['percent_complete'],
        'errorMessage': !(0, runtime_1.exists)(json, 'error_message') ? undefined : json['error_message'],
        'ingestionUnits': !(0, runtime_1.exists)(json, 'ingestion_units') ? undefined : json['ingestion_units'],
    };
}
function OperationModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'operation_type': value.operationType,
        'file_id': value.fileId,
        'status': value.status,
        'created_on': (value.createdOn.toISOString()),
        'completed_on': value.completedOn === undefined ? undefined : (value.completedOn === null ? null : value.completedOn.toISOString()),
        'percent_complete': value.percentComplete,
        'error_message': value.errorMessage,
        'ingestion_units': value.ingestionUnits,
    };
}
