// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/assistant_data/models/PaginationResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone assistant data plane API
 * Pinecone Assistant Engine is a context engine to store and retrieve relevant knowledge from millions of documents at scale. This API supports interactions with assistants.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPaginationResponse = instanceOfPaginationResponse;
exports.PaginationResponseFromJSON = PaginationResponseFromJSON;
exports.PaginationResponseFromJSONTyped = PaginationResponseFromJSONTyped;
exports.PaginationResponseToJSON = PaginationResponseToJSON;
/**
 * Check if a given object implements the PaginationResponse interface.
 */
function instanceOfPaginationResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "next" in value;
    return isInstance;
}
function PaginationResponseFromJSON(json) {
    return PaginationResponseFromJSONTyped(json, false);
}
function PaginationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'next': json['next'],
    };
}
function PaginationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'next': value.next,
    };
}
