// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch/db_control/models/BYOC2.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-04
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBYOC2 = instanceOfBYOC2;
exports.BYOC2FromJSON = BYOC2FromJSON;
exports.BYOC2FromJSONTyped = BYOC2FromJSONTyped;
exports.BYOC2ToJSON = BYOC2ToJSON;
const ByocSpecResponse_1 = require("./ByocSpecResponse");
/**
 * Check if a given object implements the BYOC2 interface.
 */
function instanceOfBYOC2(value) {
    let isInstance = true;
    isInstance = isInstance && "byoc" in value;
    return isInstance;
}
function BYOC2FromJSON(json) {
    return BYOC2FromJSONTyped(json, false);
}
function BYOC2FromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'byoc': (0, ByocSpecResponse_1.ByocSpecResponseFromJSON)(json['byoc']),
    };
}
function BYOC2ToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'byoc': (0, ByocSpecResponse_1.ByocSpecResponseToJSON)(value.byoc),
    };
}
