// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_data/apis/DocumentOperationsApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Data Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentOperationsApi = void 0;
const runtime = __importStar(require("../runtime"));
const index_1 = require("../models/index");
/**
 *
 */
class DocumentOperationsApi extends runtime.BaseAPI {
    /**
     * Delete documents from a namespace. Exactly one of `ids` or `delete_all` must be specified.  - `ids`: Delete documents with the given IDs. - `delete_all`: Delete all documents in the namespace.
     * Delete documents
     */
    async deleteDocumentsRaw(requestParameters, initOverrides) {
        if (requestParameters.xPineconeApiVersion === null || requestParameters.xPineconeApiVersion === undefined) {
            throw new runtime.RequiredError('xPineconeApiVersion', 'Required parameter requestParameters.xPineconeApiVersion was null or undefined when calling deleteDocuments.');
        }
        if (requestParameters.namespace === null || requestParameters.namespace === undefined) {
            throw new runtime.RequiredError('namespace', 'Required parameter requestParameters.namespace was null or undefined when calling deleteDocuments.');
        }
        if (requestParameters.deleteDocumentsRequest === null || requestParameters.deleteDocumentsRequest === undefined) {
            throw new runtime.RequiredError('deleteDocumentsRequest', 'Required parameter requestParameters.deleteDocumentsRequest was null or undefined when calling deleteDocuments.');
        }
        const queryParameters = {};
        const headerParameters = {};
        headerParameters['Content-Type'] = 'application/json';
        if (requestParameters.xPineconeApiVersion !== undefined && requestParameters.xPineconeApiVersion !== null) {
            headerParameters['X-Pinecone-Api-Version'] = String(requestParameters.xPineconeApiVersion);
        }
        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Api-Key"] = this.configuration.apiKey("Api-Key"); // ApiKeyAuth authentication
        }
        const response = await this.request({
            path: `/namespaces/{namespace}/documents/delete`.replace(`{${"namespace"}}`, encodeURIComponent(String(requestParameters.namespace))),
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: (0, index_1.DeleteDocumentsRequestToJSON)(requestParameters.deleteDocumentsRequest),
        }, initOverrides);
        return new runtime.JSONApiResponse(response);
    }
    /**
     * Delete documents from a namespace. Exactly one of `ids` or `delete_all` must be specified.  - `ids`: Delete documents with the given IDs. - `delete_all`: Delete all documents in the namespace.
     * Delete documents
     */
    async deleteDocuments(requestParameters, initOverrides) {
        const response = await this.deleteDocumentsRaw(requestParameters, initOverrides);
        return await response.value();
    }
    /**
     * Fetch documents from a namespace by their IDs. Returns the specified fields for each document.
     * Fetch documents
     */
    async fetchDocumentsRaw(requestParameters, initOverrides) {
        if (requestParameters.xPineconeApiVersion === null || requestParameters.xPineconeApiVersion === undefined) {
            throw new runtime.RequiredError('xPineconeApiVersion', 'Required parameter requestParameters.xPineconeApiVersion was null or undefined when calling fetchDocuments.');
        }
        if (requestParameters.namespace === null || requestParameters.namespace === undefined) {
            throw new runtime.RequiredError('namespace', 'Required parameter requestParameters.namespace was null or undefined when calling fetchDocuments.');
        }
        if (requestParameters.fetchDocumentsRequest === null || requestParameters.fetchDocumentsRequest === undefined) {
            throw new runtime.RequiredError('fetchDocumentsRequest', 'Required parameter requestParameters.fetchDocumentsRequest was null or undefined when calling fetchDocuments.');
        }
        const queryParameters = {};
        const headerParameters = {};
        headerParameters['Content-Type'] = 'application/json';
        if (requestParameters.xPineconeApiVersion !== undefined && requestParameters.xPineconeApiVersion !== null) {
            headerParameters['X-Pinecone-Api-Version'] = String(requestParameters.xPineconeApiVersion);
        }
        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Api-Key"] = this.configuration.apiKey("Api-Key"); // ApiKeyAuth authentication
        }
        const response = await this.request({
            path: `/namespaces/{namespace}/documents/fetch`.replace(`{${"namespace"}}`, encodeURIComponent(String(requestParameters.namespace))),
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: (0, index_1.FetchDocumentsRequestToJSON)(requestParameters.fetchDocumentsRequest),
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.FetchDocumentsResponseFromJSON)(jsonValue));
    }
    /**
     * Fetch documents from a namespace by their IDs. Returns the specified fields for each document.
     * Fetch documents
     */
    async fetchDocuments(requestParameters, initOverrides) {
        const response = await this.fetchDocumentsRaw(requestParameters, initOverrides);
        return await response.value();
    }
    /**
     * Search for documents in a namespace using one or more scoring methods (dense vector, sparse vector, text, or query string similarity).  Returns the top-k most similar documents along with their scores and requested fields.
     * Search documents
     */
    async searchDocumentsRaw(requestParameters, initOverrides) {
        if (requestParameters.xPineconeApiVersion === null || requestParameters.xPineconeApiVersion === undefined) {
            throw new runtime.RequiredError('xPineconeApiVersion', 'Required parameter requestParameters.xPineconeApiVersion was null or undefined when calling searchDocuments.');
        }
        if (requestParameters.namespace === null || requestParameters.namespace === undefined) {
            throw new runtime.RequiredError('namespace', 'Required parameter requestParameters.namespace was null or undefined when calling searchDocuments.');
        }
        if (requestParameters.searchDocumentsRequest === null || requestParameters.searchDocumentsRequest === undefined) {
            throw new runtime.RequiredError('searchDocumentsRequest', 'Required parameter requestParameters.searchDocumentsRequest was null or undefined when calling searchDocuments.');
        }
        const queryParameters = {};
        const headerParameters = {};
        headerParameters['Content-Type'] = 'application/json';
        if (requestParameters.xPineconeApiVersion !== undefined && requestParameters.xPineconeApiVersion !== null) {
            headerParameters['X-Pinecone-Api-Version'] = String(requestParameters.xPineconeApiVersion);
        }
        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Api-Key"] = this.configuration.apiKey("Api-Key"); // ApiKeyAuth authentication
        }
        const response = await this.request({
            path: `/namespaces/{namespace}/documents/search`.replace(`{${"namespace"}}`, encodeURIComponent(String(requestParameters.namespace))),
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: (0, index_1.SearchDocumentsRequestToJSON)(requestParameters.searchDocumentsRequest),
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.SearchDocumentsResponseFromJSON)(jsonValue));
    }
    /**
     * Search for documents in a namespace using one or more scoring methods (dense vector, sparse vector, text, or query string similarity).  Returns the top-k most similar documents along with their scores and requested fields.
     * Search documents
     */
    async searchDocuments(requestParameters, initOverrides) {
        const response = await this.searchDocumentsRaw(requestParameters, initOverrides);
        return await response.value();
    }
    /**
     * Apply partial updates to documents in a namespace.  Each update is identified by its `_id`. Any other fields set new values for those fields, and fields listed in `_remove_fields` are removed from the document. Fields that are not mentioned are left unchanged. Updates to a document that does not exist are accepted but have no effect.
     * Update documents
     */
    async updateDocumentsRaw(requestParameters, initOverrides) {
        if (requestParameters.xPineconeApiVersion === null || requestParameters.xPineconeApiVersion === undefined) {
            throw new runtime.RequiredError('xPineconeApiVersion', 'Required parameter requestParameters.xPineconeApiVersion was null or undefined when calling updateDocuments.');
        }
        if (requestParameters.namespace === null || requestParameters.namespace === undefined) {
            throw new runtime.RequiredError('namespace', 'Required parameter requestParameters.namespace was null or undefined when calling updateDocuments.');
        }
        if (requestParameters.updateDocumentsRequest === null || requestParameters.updateDocumentsRequest === undefined) {
            throw new runtime.RequiredError('updateDocumentsRequest', 'Required parameter requestParameters.updateDocumentsRequest was null or undefined when calling updateDocuments.');
        }
        const queryParameters = {};
        const headerParameters = {};
        headerParameters['Content-Type'] = 'application/json';
        if (requestParameters.xPineconeApiVersion !== undefined && requestParameters.xPineconeApiVersion !== null) {
            headerParameters['X-Pinecone-Api-Version'] = String(requestParameters.xPineconeApiVersion);
        }
        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Api-Key"] = this.configuration.apiKey("Api-Key"); // ApiKeyAuth authentication
        }
        const response = await this.request({
            path: `/namespaces/{namespace}/documents/update`.replace(`{${"namespace"}}`, encodeURIComponent(String(requestParameters.namespace))),
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: (0, index_1.UpdateDocumentsRequestToJSON)(requestParameters.updateDocumentsRequest),
        }, initOverrides);
        return new runtime.JSONApiResponse(response);
    }
    /**
     * Apply partial updates to documents in a namespace.  Each update is identified by its `_id`. Any other fields set new values for those fields, and fields listed in `_remove_fields` are removed from the document. Fields that are not mentioned are left unchanged. Updates to a document that does not exist are accepted but have no effect.
     * Update documents
     */
    async updateDocuments(requestParameters, initOverrides) {
        const response = await this.updateDocumentsRaw(requestParameters, initOverrides);
        return await response.value();
    }
    /**
     * Upsert documents into a namespace.  Each document must include an `_id` field and the fields required by the index type (for example, `_values` for dense indexes or text fields for full-text search indexes).
     * Upsert documents
     */
    async upsertDocumentsRaw(requestParameters, initOverrides) {
        if (requestParameters.xPineconeApiVersion === null || requestParameters.xPineconeApiVersion === undefined) {
            throw new runtime.RequiredError('xPineconeApiVersion', 'Required parameter requestParameters.xPineconeApiVersion was null or undefined when calling upsertDocuments.');
        }
        if (requestParameters.namespace === null || requestParameters.namespace === undefined) {
            throw new runtime.RequiredError('namespace', 'Required parameter requestParameters.namespace was null or undefined when calling upsertDocuments.');
        }
        if (requestParameters.upsertDocumentsRequest === null || requestParameters.upsertDocumentsRequest === undefined) {
            throw new runtime.RequiredError('upsertDocumentsRequest', 'Required parameter requestParameters.upsertDocumentsRequest was null or undefined when calling upsertDocuments.');
        }
        const queryParameters = {};
        const headerParameters = {};
        headerParameters['Content-Type'] = 'application/json';
        if (requestParameters.xPineconeApiVersion !== undefined && requestParameters.xPineconeApiVersion !== null) {
            headerParameters['X-Pinecone-Api-Version'] = String(requestParameters.xPineconeApiVersion);
        }
        if (this.configuration && this.configuration.apiKey) {
            headerParameters["Api-Key"] = this.configuration.apiKey("Api-Key"); // ApiKeyAuth authentication
        }
        const response = await this.request({
            path: `/namespaces/{namespace}/documents/upsert`.replace(`{${"namespace"}}`, encodeURIComponent(String(requestParameters.namespace))),
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: (0, index_1.UpsertDocumentsRequestToJSON)(requestParameters.upsertDocumentsRequest),
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, index_1.UpsertDocumentsResponseFromJSON)(jsonValue));
    }
    /**
     * Upsert documents into a namespace.  Each document must include an `_id` field and the fields required by the index type (for example, `_values` for dense indexes or text fields for full-text search indexes).
     * Upsert documents
     */
    async upsertDocuments(requestParameters, initOverrides) {
        const response = await this.upsertDocumentsRaw(requestParameters, initOverrides);
        return await response.value();
    }
}
exports.DocumentOperationsApi = DocumentOperationsApi;
