// @pinecone-database/pinecone/dist/integration/preview-control/configureIndex.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const index_1 = require("../../index");
const test_helpers_1 = require("../test-helpers");
let pc;
beforeAll(() => {
    pc = new index_1.Pinecone();
});
describe('preview configureIndex', () => {
    test('updates deletionProtection on an existing index', async () => {
        const indexName = (0, test_helpers_1.randomName)('preview-cfg');
        await pc.preview.indexes.create({
            name: indexName,
            schema: {
                fields: {
                    chunk_text: { type: 'string', fullTextSearch: {} },
                },
            },
            waitUntilReady: true,
        });
        const updated = await pc.preview.indexes.configure(indexName, {
            deletionProtection: 'enabled',
        });
        expect(updated.deletionProtection).toBe('enabled');
        // Disable protection before deletion
        await pc.preview.indexes.configure(indexName, {
            deletionProtection: 'disabled',
        });
        await pc.preview.indexes.delete(indexName);
    });
    test('throws on configuring a non-existent index', async () => {
        await expect(pc.preview.indexes.configure((0, test_helpers_1.randomName)('no-such-idx'), {
            deletionProtection: 'enabled',
        })).rejects.toThrow();
    });
});
