// @pinecone-database/pinecone/dist/preview/indexes/__tests__/createIndex.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const createIndex_1 = require("../createIndex");
const errors_1 = require("../../../errors");
const validOptions = {
    name: 'schema-idx',
    schema: {
        fields: {
            chunk_text: { type: 'string', fullTextSearch: {} },
        },
    },
};
const buildMockApi = (overrides = {}) => ({
    createIndex: jest.fn().mockResolvedValue({
        name: 'schema-idx',
        host: 'schema-idx.svc.pinecone.io',
        schema: {
            fields: { chunk_text: { type: 'string', fullTextSearch: {} } },
        },
        status: { ready: true, state: 'Ready' },
        deployment: { deployment_type: 'managed' },
        deletion_protection: 'disabled',
    }),
    describeIndex: jest.fn(),
    ...overrides,
});
describe('createPreviewIndex', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when name is empty string', async () => {
            await expect((0, createIndex_1.createPreviewIndex)(buildMockApi(), { ...validOptions, name: '' })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references the `name` field', async () => {
            await expect((0, createIndex_1.createPreviewIndex)(buildMockApi(), { ...validOptions, name: '' })).rejects.toThrow('`name`');
        });
        test('throws PineconeArgumentError when schema is missing', async () => {
            await expect((0, createIndex_1.createPreviewIndex)(buildMockApi(), {
                name: 'my-index',
            })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references the `schema` field', async () => {
            await expect((0, createIndex_1.createPreviewIndex)(buildMockApi(), {
                name: 'my-index',
            })).rejects.toThrow('`schema`');
        });
    });
    describe('success', () => {
        test('returns IndexModel on success', async () => {
            const result = await (0, createIndex_1.createPreviewIndex)(buildMockApi(), validOptions);
            expect(result).toMatchObject({ name: 'schema-idx' });
        });
        test('passes name to the API client', async () => {
            const api = buildMockApi();
            await (0, createIndex_1.createPreviewIndex)(api, validOptions);
            expect(api.createIndex).toHaveBeenCalledWith(expect.objectContaining({
                createIndexRequest: expect.objectContaining({ name: 'schema-idx' }),
            }));
        });
        test('passes schema to the API client', async () => {
            const api = buildMockApi();
            await (0, createIndex_1.createPreviewIndex)(api, validOptions);
            expect(api.createIndex).toHaveBeenCalledWith(expect.objectContaining({
                createIndexRequest: expect.objectContaining({
                    schema: validOptions.schema,
                }),
            }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, createIndex_1.createPreviewIndex)(api, validOptions);
            expect(api.createIndex).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
        test('does not pass waitUntilReady to the API client', async () => {
            const api = buildMockApi();
            await (0, createIndex_1.createPreviewIndex)(api, { ...validOptions, waitUntilReady: false });
            expect(api.createIndex).toHaveBeenCalledWith(expect.objectContaining({
                createIndexRequest: expect.not.objectContaining({
                    waitUntilReady: expect.anything(),
                }),
            }));
        });
        test('polls with describeIndex when waitUntilReady is true', async () => {
            const api = buildMockApi({
                describeIndex: jest.fn().mockResolvedValue({
                    name: 'schema-idx',
                    host: 'schema-idx.svc.pinecone.io',
                    schema: { fields: {} },
                    status: { ready: true, state: 'Ready' },
                    deployment: { deployment_type: 'managed' },
                    deletion_protection: 'disabled',
                }),
            });
            await (0, createIndex_1.createPreviewIndex)(api, { ...validOptions, waitUntilReady: true });
            expect(api.describeIndex).toHaveBeenCalledWith(expect.objectContaining({ indexName: 'schema-idx' }));
        });
    });
    describe('waitUntilReady polling', () => {
        beforeEach(() => jest.useFakeTimers());
        afterEach(() => jest.useRealTimers());
        test('resolves once index becomes ready after not-ready polls', async () => {
            const api = buildMockApi({
                describeIndex: jest
                    .fn()
                    .mockResolvedValueOnce({
                    status: { ready: false, state: 'Initializing' },
                })
                    .mockResolvedValueOnce({ status: { ready: true, state: 'Ready' } }),
            });
            const promise = (0, createIndex_1.createPreviewIndex)(api, {
                ...validOptions,
                waitUntilReady: true,
            });
            await jest.advanceTimersByTimeAsync(10000);
            const result = await promise;
            expect(result.status?.state).toBe('Ready');
            expect(api.describeIndex).toHaveBeenCalledTimes(2);
        });
        test('throws PineconeIndexInitializationFailedError on InitializationFailed state', async () => {
            const api = buildMockApi({
                describeIndex: jest.fn().mockResolvedValue({
                    status: { ready: false, state: 'InitializationFailed' },
                }),
            });
            await expect((0, createIndex_1.createPreviewIndex)(api, { ...validOptions, waitUntilReady: true })).rejects.toThrow(errors_1.PineconeIndexInitializationFailedError);
        });
        test('throws PineconeIndexTerminatedError on Terminating state', async () => {
            const api = buildMockApi({
                describeIndex: jest.fn().mockResolvedValue({
                    status: { ready: false, state: 'Terminating' },
                }),
            });
            await expect((0, createIndex_1.createPreviewIndex)(api, { ...validOptions, waitUntilReady: true })).rejects.toThrow(errors_1.PineconeIndexTerminatedError);
        });
        test('throws PineconeIndexTerminatedError on Disabled state', async () => {
            const api = buildMockApi({
                describeIndex: jest
                    .fn()
                    .mockResolvedValue({ status: { ready: false, state: 'Disabled' } }),
            });
            await expect((0, createIndex_1.createPreviewIndex)(api, { ...validOptions, waitUntilReady: true })).rejects.toThrow(errors_1.PineconeIndexTerminatedError);
        });
        test('throws PineconeTimeoutError when timeout elapses', async () => {
            const api = buildMockApi({
                describeIndex: jest.fn().mockResolvedValue({
                    status: { ready: false, state: 'Initializing' },
                }),
            });
            const promise = (0, createIndex_1.createPreviewIndex)(api, {
                ...validOptions,
                waitUntilReady: true,
                timeout: 8000,
            });
            // Attach rejection handler before advancing timers to avoid unhandled rejection warning
            const assertion = expect(promise).rejects.toThrow(errors_1.PineconeTimeoutError);
            await jest.advanceTimersByTimeAsync(10000);
            await assertion;
        });
    });
    describe('error handling', () => {
        test('converts ResponseError to a typed Pinecone error', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_control')));
            const api = buildMockApi({
                createIndex: jest.fn().mockRejectedValue(new ResponseError(new Response('{"error": "Index already exists"}', {
                    status: 409,
                }), 'Conflict')),
            });
            await expect((0, createIndex_1.createPreviewIndex)(api, validOptions)).rejects.toBeDefined();
        });
    });
});
