// @pinecone-database/pinecone/dist/preview/indexes/__tests__/createCollection.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const createCollection_1 = require("../createCollection");
const buildMockApi = (overrides = {}) => ({
    createCollection: jest.fn().mockResolvedValue({
        name: 'test-collection',
        status: 'Initializing',
        environment: 'us-east1-gcp',
        dimension: 3,
    }),
    ...overrides,
});
describe('createPreviewCollection', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when name is empty', async () => {
            await expect((0, createCollection_1.createPreviewCollection)(buildMockApi(), {
                name: '',
                source: 'my-index',
            })).rejects.toThrow('`name`');
        });
        test('throws PineconeArgumentError when source is empty', async () => {
            await expect((0, createCollection_1.createPreviewCollection)(buildMockApi(), { name: 'col', source: '' })).rejects.toThrow('`source`');
        });
    });
    describe('success', () => {
        test('returns CollectionModel on success', async () => {
            const result = await (0, createCollection_1.createPreviewCollection)(buildMockApi(), {
                name: 'test-collection',
                source: 'my-index',
            });
            expect(result).toMatchObject({
                name: 'test-collection',
                status: 'Initializing',
            });
        });
        test('calls api.createCollection with alpha API version header', async () => {
            const api = buildMockApi();
            await (0, createCollection_1.createPreviewCollection)(api, {
                name: 'test-collection',
                source: 'my-index',
            });
            expect(api.createCollection).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
        test('passes name and source in the request body', async () => {
            const api = buildMockApi();
            await (0, createCollection_1.createPreviewCollection)(api, {
                name: 'my-col',
                source: 'my-index',
            });
            expect(api.createCollection).toHaveBeenCalledWith(expect.objectContaining({
                createCollectionRequest: { name: 'my-col', source: 'my-index' },
            }));
        });
    });
    describe('error handling', () => {
        test('converts 409 ResponseError to a thrown error', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_control')));
            const api = buildMockApi({
                createCollection: jest
                    .fn()
                    .mockRejectedValue(new ResponseError(new Response('Already exists', { status: 409 }), 'Already exists')),
            });
            await expect((0, createCollection_1.createPreviewCollection)(api, { name: 'my-col', source: 'my-index' })).rejects.toBeDefined();
        });
    });
});
