// @pinecone-database/pinecone/dist/preview/indexes/__tests__/createIndexFromBackup.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createIndexFromBackup_1 = require("../createIndexFromBackup");
const errors_1 = require("../../../errors");
const mockResponse = { restore_job_id: 'rj-abc123', index_id: 'idx-xyz789' };
const buildMockApi = (overrides = {}) => ({
    createIndexFromBackupOperation: jest.fn().mockResolvedValue(mockResponse),
    ...overrides,
});
describe('createIndexFromBackup', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when backupId is empty', async () => {
            await expect((0, createIndexFromBackup_1.createIndexFromBackup)(buildMockApi(), '', { name: 'my-index' })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references `backupId` field', async () => {
            await expect((0, createIndexFromBackup_1.createIndexFromBackup)(buildMockApi(), '', { name: 'my-index' })).rejects.toThrow('`backupId`');
        });
        test('throws PineconeArgumentError when name is empty', async () => {
            await expect((0, createIndexFromBackup_1.createIndexFromBackup)(buildMockApi(), 'bk-123', { name: '' })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references `name` field', async () => {
            await expect((0, createIndexFromBackup_1.createIndexFromBackup)(buildMockApi(), 'bk-123', { name: '' })).rejects.toThrow('`name`');
        });
    });
    describe('success', () => {
        test('returns CreateIndexFromBackupResponse on success', async () => {
            const result = await (0, createIndexFromBackup_1.createIndexFromBackup)(buildMockApi(), 'bk-abc', {
                name: 'restored-index',
            });
            expect(result).toEqual(mockResponse);
        });
        test('passes backupId and name to the API client', async () => {
            const api = buildMockApi();
            await (0, createIndexFromBackup_1.createIndexFromBackup)(api, 'bk-abc', { name: 'restored-index' });
            expect(api.createIndexFromBackupOperation).toHaveBeenCalledWith(expect.objectContaining({
                backupId: 'bk-abc',
                createIndexFromBackupRequest: expect.objectContaining({
                    name: 'restored-index',
                }),
            }));
        });
        test('passes optional tags to the API client when provided', async () => {
            const api = buildMockApi();
            await (0, createIndexFromBackup_1.createIndexFromBackup)(api, 'bk-abc', {
                name: 'restored-index',
                tags: { env: 'test' },
            });
            expect(api.createIndexFromBackupOperation).toHaveBeenCalledWith(expect.objectContaining({
                createIndexFromBackupRequest: expect.objectContaining({
                    tags: { env: 'test' },
                }),
            }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, createIndexFromBackup_1.createIndexFromBackup)(api, 'bk-abc', { name: 'restored-index' });
            expect(api.createIndexFromBackupOperation).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('propagates errors from the API client', async () => {
            const api = buildMockApi({
                createIndexFromBackupOperation: jest
                    .fn()
                    .mockRejectedValue(new Response('Not Found', { status: 404 })),
            });
            await expect((0, createIndexFromBackup_1.createIndexFromBackup)(api, 'bk-missing', { name: 'idx' })).rejects.toBeDefined();
        });
    });
});
