// @pinecone-database/pinecone/dist/preview/indexes/__tests__/createBackup.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createBackup_1 = require("../createBackup");
const errors_1 = require("../../../errors");
const buildMockApi = (overrides = {}) => ({
    createBackup: jest.fn().mockResolvedValue({
        backupId: 'bk-123',
        sourceIndexName: 'my-index',
        sourceIndexId: 'idx-abc',
        status: 'Initializing',
        cloud: 'aws',
        region: 'us-east-1',
    }),
    ...overrides,
});
describe('createPreviewBackup', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when indexName is empty string', async () => {
            await expect((0, createBackup_1.createPreviewBackup)(buildMockApi(), '')).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references the `indexName` field', async () => {
            await expect((0, createBackup_1.createPreviewBackup)(buildMockApi(), '')).rejects.toThrow('`indexName`');
        });
    });
    describe('success', () => {
        test('returns BackupModel on success', async () => {
            const result = await (0, createBackup_1.createPreviewBackup)(buildMockApi(), 'my-index');
            expect(result).toMatchObject({
                backupId: 'bk-123',
                status: 'Initializing',
            });
        });
        test('passes indexName to the API', async () => {
            const api = buildMockApi();
            await (0, createBackup_1.createPreviewBackup)(api, 'my-index');
            expect(api.createBackup).toHaveBeenCalledWith(expect.objectContaining({ indexName: 'my-index' }));
        });
        test('passes name and description when provided', async () => {
            const api = buildMockApi();
            await (0, createBackup_1.createPreviewBackup)(api, 'my-index', {
                name: 'snap',
                description: 'nightly',
            });
            expect(api.createBackup).toHaveBeenCalledWith(expect.objectContaining({
                createBackupRequest: { name: 'snap', description: 'nightly' },
            }));
        });
        test('passes empty createBackupRequest when options omitted', async () => {
            const api = buildMockApi();
            await (0, createBackup_1.createPreviewBackup)(api, 'my-index');
            expect(api.createBackup).toHaveBeenCalledWith(expect.objectContaining({ createBackupRequest: {} }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, createBackup_1.createPreviewBackup)(api, 'my-index');
            expect(api.createBackup).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('wraps API errors via handleApiError', async () => {
            const api = buildMockApi({
                createBackup: jest.fn().mockRejectedValue(Object.assign(new Error(), {
                    status: 404,
                    text: async () => 'Index not found',
                })),
            });
            await expect((0, createBackup_1.createPreviewBackup)(api, 'my-index')).rejects.toThrow();
        });
        test('does not re-wrap PineconeBadRequestError from middleware as PineconeConnectionError', async () => {
            const badRequestError = new errors_1.PineconeBadRequestError({
                status: 400,
                message: 'Backup creation is not supported for this index',
            });
            const api = buildMockApi({
                createBackup: jest.fn().mockRejectedValue(badRequestError),
            });
            await expect((0, createBackup_1.createPreviewBackup)(api, 'my-index')).rejects.toThrow(errors_1.PineconeBadRequestError);
            await expect((0, createBackup_1.createPreviewBackup)(api, 'my-index')).rejects.not.toThrow(errors_1.PineconeConnectionError);
        });
    });
});
