// @pinecone-database/pinecone/dist/preview/indexes/__tests__/deleteDocuments.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const deleteDocuments_1 = require("../deleteDocuments");
const errors_1 = require("../../../errors");
const buildMockApi = (overrides = {}) => ({
    deleteDocuments: jest.fn().mockResolvedValue({}),
    ...overrides,
});
describe('deletePreviewDocuments', () => {
    describe('validation', () => {
        test('throws when neither ids nor deleteAll is provided', async () => {
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', {})).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references ids and deleteAll when neither is provided', async () => {
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', {})).rejects.toThrow('`ids` or `deleteAll`');
        });
        test('throws when both ids and deleteAll are provided', async () => {
            const options = {
                ids: ['doc-1'],
                deleteAll: true,
            };
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', options)).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message mentions mutual exclusion when both ids and deleteAll are set', async () => {
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', {
                ids: ['doc-1'],
                deleteAll: true,
            })).rejects.toThrow('mutually exclusive');
        });
        test('throws when ids is an empty array', async () => {
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', { ids: [] })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references ids when the array is empty', async () => {
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', { ids: [] })).rejects.toThrow('`ids`');
        });
    });
    describe('success', () => {
        test('resolves to void when ids are provided', async () => {
            const result = await (0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', {
                ids: ['doc-1', 'doc-2'],
            });
            expect(result).toBeUndefined();
        });
        test('passes ids in deleteDocumentsRequest', async () => {
            const api = buildMockApi();
            await (0, deleteDocuments_1.deletePreviewDocuments)(api, 'test-ns', { ids: ['doc-1', 'doc-2'] });
            expect(api.deleteDocuments).toHaveBeenCalledWith(expect.objectContaining({
                deleteDocumentsRequest: expect.objectContaining({
                    ids: ['doc-1', 'doc-2'],
                }),
            }));
        });
        test('resolves to void when deleteAll is true', async () => {
            const result = await (0, deleteDocuments_1.deletePreviewDocuments)(buildMockApi(), 'test-ns', {
                deleteAll: true,
            });
            expect(result).toBeUndefined();
        });
        test('passes deleteAll in deleteDocumentsRequest', async () => {
            const api = buildMockApi();
            await (0, deleteDocuments_1.deletePreviewDocuments)(api, 'test-ns', { deleteAll: true });
            expect(api.deleteDocuments).toHaveBeenCalledWith(expect.objectContaining({
                deleteDocumentsRequest: expect.objectContaining({
                    deleteAll: true,
                }),
            }));
        });
        test('passes namespace to the API call', async () => {
            const api = buildMockApi();
            await (0, deleteDocuments_1.deletePreviewDocuments)(api, 'my-namespace', {
                ids: ['doc-1'],
            });
            expect(api.deleteDocuments).toHaveBeenCalledWith(expect.objectContaining({ namespace: 'my-namespace' }));
        });
        test('passes xPineconeApiVersion containing 2026 in every call', async () => {
            const api = buildMockApi();
            await (0, deleteDocuments_1.deletePreviewDocuments)(api, 'test-ns', { ids: ['doc-1'] });
            expect(api.deleteDocuments).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('converts ResponseError to a typed Pinecone error via handleApiError', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_data')));
            const api = buildMockApi({
                deleteDocuments: jest.fn().mockRejectedValue(new ResponseError(new Response('{"code": 5, "message": "Index not found"}', {
                    status: 404,
                }), 'Not Found')),
            });
            await expect((0, deleteDocuments_1.deletePreviewDocuments)(api, 'test-ns', { ids: ['doc-1'] })).rejects.toBeDefined();
        });
    });
});
