// @pinecone-database/pinecone/dist/preview/indexes/__tests__/describeRestoreJob.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const describeRestoreJob_1 = require("../describeRestoreJob");
const errors_1 = require("../../../errors");
const mockRestoreJob = {
    restoreJobId: 'rj-abc123',
    backupId: 'bk-xyz',
    targetIndexName: 'my-restored-index',
    targetIndexId: 'idx-111',
    status: 'Completed',
    percentComplete: 100,
};
const buildMockApi = (overrides = {}) => ({
    describeRestoreJob: jest.fn().mockResolvedValue(mockRestoreJob),
    ...overrides,
});
describe('describePreviewRestoreJob', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when jobId is empty string', async () => {
            await expect((0, describeRestoreJob_1.describePreviewRestoreJob)(buildMockApi(), '')).rejects.toThrow(errors_1.PineconeArgumentError);
        });
    });
    describe('success', () => {
        test('returns the RestoreJobModel', async () => {
            const result = await (0, describeRestoreJob_1.describePreviewRestoreJob)(buildMockApi(), 'rj-abc123');
            expect(result).toEqual(mockRestoreJob);
        });
        test('calls describeRestoreJob once with the correct jobId', async () => {
            const api = buildMockApi();
            await (0, describeRestoreJob_1.describePreviewRestoreJob)(api, 'rj-abc123');
            expect(api.describeRestoreJob).toHaveBeenCalledTimes(1);
            expect(api.describeRestoreJob).toHaveBeenCalledWith(expect.objectContaining({ jobId: 'rj-abc123' }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, describeRestoreJob_1.describePreviewRestoreJob)(api, 'rj-abc123');
            expect(api.describeRestoreJob).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('wraps API errors via handleApiError', async () => {
            const api = buildMockApi({
                describeRestoreJob: jest
                    .fn()
                    .mockRejectedValue(new Response('Internal Server Error', { status: 500 })),
            });
            await expect((0, describeRestoreJob_1.describePreviewRestoreJob)(api, 'rj-abc123')).rejects.toBeDefined();
        });
    });
});
