// @pinecone-database/pinecone/dist/preview/indexes/__tests__/describeBackup.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const describeBackup_1 = require("../describeBackup");
const errors_1 = require("../../../errors");
const buildMockApi = (overrides = {}) => ({
    describeBackup: jest.fn().mockResolvedValue({
        backupId: 'bk-123',
        sourceIndexName: 'my-index',
        sourceIndexId: 'idx-abc',
        status: 'Ready',
        cloud: 'aws',
        region: 'us-east-1',
    }),
    ...overrides,
});
describe('describePreviewBackup', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when backupId is empty', async () => {
            await expect((0, describeBackup_1.describePreviewBackup)(buildMockApi(), '')).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references the `backupId` field', async () => {
            await expect((0, describeBackup_1.describePreviewBackup)(buildMockApi(), '')).rejects.toThrow('`backupId`');
        });
    });
    describe('success', () => {
        test('returns BackupModel on success', async () => {
            const result = await (0, describeBackup_1.describePreviewBackup)(buildMockApi(), 'bk-123');
            expect(result).toMatchObject({ backupId: 'bk-123', status: 'Ready' });
        });
        test('passes backupId to the API client', async () => {
            const api = buildMockApi();
            await (0, describeBackup_1.describePreviewBackup)(api, 'bk-abc');
            expect(api.describeBackup).toHaveBeenCalledWith(expect.objectContaining({ backupId: 'bk-abc' }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, describeBackup_1.describePreviewBackup)(api, 'bk-123');
            expect(api.describeBackup).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('propagates errors from the API client', async () => {
            const api = buildMockApi({
                describeBackup: jest
                    .fn()
                    .mockRejectedValue(new Response('Not Found', { status: 404 })),
            });
            await expect((0, describeBackup_1.describePreviewBackup)(api, 'bk-missing')).rejects.toBeDefined();
        });
    });
});
