// @pinecone-database/pinecone/dist/preview/indexes/__tests__/describeCollection.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const describeCollection_1 = require("../describeCollection");
const buildMockApi = (overrides = {}) => ({
    describeCollection: jest.fn().mockResolvedValue({
        name: 'my-collection',
        status: 'Ready',
        environment: 'us-east1-gcp',
        size: 3126700,
        dimension: 3,
        vector_count: 99,
    }),
    ...overrides,
});
describe('describePreviewCollection', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when collectionName is empty', async () => {
            await expect((0, describeCollection_1.describePreviewCollection)(buildMockApi(), '')).rejects.toThrow('`collectionName`');
        });
    });
    describe('success', () => {
        test('returns CollectionModel on success', async () => {
            const result = await (0, describeCollection_1.describePreviewCollection)(buildMockApi(), 'my-collection');
            expect(result).toMatchObject({
                name: 'my-collection',
                status: 'Ready',
                environment: 'us-east1-gcp',
            });
        });
        test('calls api.describeCollection with the correct collectionName', async () => {
            const api = buildMockApi();
            await (0, describeCollection_1.describePreviewCollection)(api, 'my-collection');
            expect(api.describeCollection).toHaveBeenCalledWith(expect.objectContaining({ collectionName: 'my-collection' }));
        });
        test('calls api.describeCollection with alpha API version header', async () => {
            const api = buildMockApi();
            await (0, describeCollection_1.describePreviewCollection)(api, 'my-collection');
            expect(api.describeCollection).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('converts 404 ResponseError to a thrown error', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_control')));
            const api = buildMockApi({
                describeCollection: jest
                    .fn()
                    .mockRejectedValue(new ResponseError(new Response('Not Found', { status: 404 }), 'Not Found')),
            });
            await expect((0, describeCollection_1.describePreviewCollection)(api, 'missing-collection')).rejects.toBeDefined();
        });
    });
});
