// @pinecone-database/pinecone/dist/preview/indexes/__tests__/describeIndex.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const describeIndex_1 = require("../describeIndex");
const errors_1 = require("../../../errors");
const mockIndexModel = {
    name: 'my-index',
    host: 'my-index.svc.pinecone.io',
    status: { ready: true, state: 'Ready' },
    schema: { fields: { genre: { type: 'string' } } },
    deployment: { deployment_type: 'managed' },
    deletion_protection: 'disabled',
};
const buildMockApi = (overrides = {}) => ({
    describeIndex: jest.fn().mockResolvedValue(mockIndexModel),
    ...overrides,
});
describe('describePreviewIndex', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when indexName is empty string', async () => {
            await expect((0, describeIndex_1.describePreviewIndex)(buildMockApi(), '')).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references the `indexName` field', async () => {
            await expect((0, describeIndex_1.describePreviewIndex)(buildMockApi(), '')).rejects.toThrow('`indexName`');
        });
    });
    describe('success', () => {
        test('returns IndexModel on success', async () => {
            const result = await (0, describeIndex_1.describePreviewIndex)(buildMockApi(), 'my-index');
            expect(result).toMatchObject({ name: 'my-index' });
        });
        test('passes indexName to the API client', async () => {
            const api = buildMockApi();
            await (0, describeIndex_1.describePreviewIndex)(api, 'my-index');
            expect(api.describeIndex).toHaveBeenCalledWith(expect.objectContaining({ indexName: 'my-index' }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, describeIndex_1.describePreviewIndex)(api, 'my-index');
            expect(api.describeIndex).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
        test('returns schema field when index has a schema', async () => {
            const result = await (0, describeIndex_1.describePreviewIndex)(buildMockApi(), 'my-index');
            expect(result.schema).toBeDefined();
        });
    });
    describe('error handling', () => {
        test('rethrows error from handleApiError', async () => {
            const api = buildMockApi({
                describeIndex: jest.fn().mockRejectedValue(new Error('Network error')),
            });
            await expect((0, describeIndex_1.describePreviewIndex)(api, 'my-index')).rejects.toBeDefined();
        });
        test('converts ResponseError to a typed Pinecone error', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_control')));
            const api = buildMockApi({
                describeIndex: jest
                    .fn()
                    .mockRejectedValue(new ResponseError(new Response('{"error": "Index not found"}', { status: 404 }), 'Not Found')),
            });
            await expect((0, describeIndex_1.describePreviewIndex)(api, 'missing-index')).rejects.toBeDefined();
        });
    });
});
