// @pinecone-database/pinecone/dist/preview/indexes/__tests__/fetchDocuments.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const fetchDocuments_1 = require("../fetchDocuments");
const errors_1 = require("../../../errors");
const mockResponse = {
    documents: {
        'doc-1': { _id: 'doc-1', title: 'ML Doc', content: 'Machine learning' },
        'doc-2': { _id: 'doc-2', title: 'VDB Doc', content: 'Vector databases' },
    },
    namespace: 'test-ns',
    usage: { readUnits: 2 },
};
const buildMockApi = (overrides = {}) => ({
    fetchDocuments: jest.fn().mockResolvedValue(mockResponse),
    ...overrides,
});
const validOptions = {
    ids: ['doc-1', 'doc-2'],
};
describe('fetchPreviewDocuments', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when ids is an empty array', async () => {
            await expect((0, fetchDocuments_1.fetchPreviewDocuments)(buildMockApi(), 'test-ns', { ids: [] })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references `ids` when the array is empty', async () => {
            await expect((0, fetchDocuments_1.fetchPreviewDocuments)(buildMockApi(), 'test-ns', { ids: [] })).rejects.toThrow('`ids`');
        });
        test('throws PineconeArgumentError when ids is missing', async () => {
            await expect((0, fetchDocuments_1.fetchPreviewDocuments)(buildMockApi(), 'test-ns', {})).rejects.toThrow(errors_1.PineconeArgumentError);
        });
    });
    describe('success', () => {
        test('returns PreviewFetchDocumentsResponse with documents, namespace, usage', async () => {
            const result = await (0, fetchDocuments_1.fetchPreviewDocuments)(buildMockApi(), 'test-ns', validOptions);
            expect(result.documents).toBeDefined();
            expect(result.namespace).toBe('test-ns');
            expect(result.usage.readUnits).toBeGreaterThanOrEqual(0);
        });
        test('passes namespace to the API call', async () => {
            const api = buildMockApi();
            await (0, fetchDocuments_1.fetchPreviewDocuments)(api, 'my-namespace', validOptions);
            expect(api.fetchDocuments).toHaveBeenCalledWith(expect.objectContaining({ namespace: 'my-namespace' }));
        });
        test('passes ids to the API call', async () => {
            const api = buildMockApi();
            await (0, fetchDocuments_1.fetchPreviewDocuments)(api, 'test-ns', validOptions);
            expect(api.fetchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                fetchDocumentsRequest: expect.objectContaining({
                    ids: ['doc-1', 'doc-2'],
                }),
            }));
        });
        test('passes includeFields when provided', async () => {
            const api = buildMockApi();
            await (0, fetchDocuments_1.fetchPreviewDocuments)(api, 'test-ns', {
                ...validOptions,
                includeFields: ['title'],
            });
            expect(api.fetchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                fetchDocumentsRequest: expect.objectContaining({
                    includeFields: ['title'],
                }),
            }));
        });
        test('passes xPineconeApiVersion containing 2026 in every call', async () => {
            const api = buildMockApi();
            await (0, fetchDocuments_1.fetchPreviewDocuments)(api, 'test-ns', validOptions);
            expect(api.fetchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('converts ResponseError (404) to a typed Pinecone error via handleApiError', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_data')));
            const api = buildMockApi({
                fetchDocuments: jest.fn().mockRejectedValue(new ResponseError(new Response('{"code": 5, "message": "Index not found"}', {
                    status: 404,
                }), 'Not Found')),
            });
            await expect((0, fetchDocuments_1.fetchPreviewDocuments)(api, 'test-ns', validOptions)).rejects.toBeDefined();
        });
    });
});
