// @pinecone-database/pinecone/dist/preview/indexes/__tests__/listCollections.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const listCollections_1 = require("../listCollections");
const buildMockApi = (overrides = {}) => ({
    listCollections: jest.fn().mockResolvedValue({
        collections: [
            {
                name: 'col-a',
                status: 'Ready',
                dimension: 3,
                environment: 'us-east1-gcp',
            },
        ],
    }),
    ...overrides,
});
describe('listPreviewCollections', () => {
    describe('success', () => {
        test('returns CollectionList with collections array', async () => {
            const result = await (0, listCollections_1.listPreviewCollections)(buildMockApi());
            expect(result).toHaveProperty('collections');
            expect(Array.isArray(result.collections)).toBe(true);
        });
        test('returns empty collections array when no collections exist', async () => {
            const api = buildMockApi({
                listCollections: jest.fn().mockResolvedValue({ collections: [] }),
            });
            const result = await (0, listCollections_1.listPreviewCollections)(api);
            expect(result).toEqual({ collections: [] });
        });
        test('calls api.listCollections with alpha API version header', async () => {
            const api = buildMockApi();
            await (0, listCollections_1.listPreviewCollections)(api);
            expect(api.listCollections).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('converts 401 ResponseError to a typed Pinecone error', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_control')));
            const api = buildMockApi({
                listCollections: jest
                    .fn()
                    .mockRejectedValue(new ResponseError(new Response('Unauthorized', { status: 401 }), 'Unauthorized')),
            });
            await expect((0, listCollections_1.listPreviewCollections)(api)).rejects.toBeDefined();
        });
    });
});
