// @pinecone-database/pinecone/dist/preview/indexes/__tests__/listProjectBackups.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const listProjectBackups_1 = require("../listProjectBackups");
const buildMockApi = (overrides = {}) => ({
    listProjectBackups: jest.fn().mockResolvedValue({
        data: [
            {
                backupId: 'bk-proj-1',
                sourceIndexName: 'index-a',
                sourceIndexId: 'idx-001',
                status: 'Ready',
                cloud: 'aws',
                region: 'us-east-1',
            },
            {
                backupId: 'bk-proj-2',
                sourceIndexName: 'index-b',
                sourceIndexId: 'idx-002',
                status: 'Ready',
                cloud: 'aws',
                region: 'us-west-2',
            },
        ],
        pagination: { next: undefined },
    }),
    ...overrides,
});
describe('listPreviewProjectBackups', () => {
    describe('success', () => {
        test('returns BackupList with data array', async () => {
            const result = await (0, listProjectBackups_1.listPreviewProjectBackups)(buildMockApi());
            expect(Array.isArray(result.data)).toBe(true);
            expect(result.data).toHaveLength(2);
        });
        test('returns backup entries spanning multiple source indexes', async () => {
            const result = await (0, listProjectBackups_1.listPreviewProjectBackups)(buildMockApi());
            const names = result.data?.map((b) => b.sourceIndexName);
            expect(names).toContain('index-a');
            expect(names).toContain('index-b');
        });
        test('calls listProjectBackups with no required path args', async () => {
            const api = buildMockApi();
            await (0, listProjectBackups_1.listPreviewProjectBackups)(api);
            expect(api.listProjectBackups).toHaveBeenCalledTimes(1);
        });
        test('passes limit and paginationToken when provided', async () => {
            const api = buildMockApi();
            const options = {
                limit: 5,
                paginationToken: 'tok-abc',
            };
            await (0, listProjectBackups_1.listPreviewProjectBackups)(api, options);
            expect(api.listProjectBackups).toHaveBeenCalledWith(expect.objectContaining({ limit: 5, paginationToken: 'tok-abc' }));
        });
        test('omits limit and paginationToken when options not provided', async () => {
            const api = buildMockApi();
            await (0, listProjectBackups_1.listPreviewProjectBackups)(api);
            expect(api.listProjectBackups).toHaveBeenCalledWith(expect.objectContaining({
                limit: undefined,
                paginationToken: undefined,
            }));
        });
        test('uses alpha API version header', async () => {
            const api = buildMockApi();
            await (0, listProjectBackups_1.listPreviewProjectBackups)(api);
            expect(api.listProjectBackups).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('pagination', () => {
        test('returns next pagination token when present', async () => {
            const api = buildMockApi({
                listProjectBackups: jest.fn().mockResolvedValue({
                    data: [],
                    pagination: { next: 'next-page-token' },
                }),
            });
            const result = await (0, listProjectBackups_1.listPreviewProjectBackups)(api);
            expect(result.pagination?.next).toEqual('next-page-token');
        });
    });
    describe('error handling', () => {
        test('wraps API errors via handleApiError', async () => {
            const api = buildMockApi({
                listProjectBackups: jest
                    .fn()
                    .mockRejectedValue(new Response('Internal Server Error', { status: 500 })),
            });
            await expect((0, listProjectBackups_1.listPreviewProjectBackups)(api)).rejects.toBeDefined();
        });
    });
});
