// @pinecone-database/pinecone/dist/preview/indexes/__tests__/searchDocuments.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const searchDocuments_1 = require("../searchDocuments");
const errors_1 = require("../../../errors");
const mockResponse = {
    matches: [
        { _id: 'doc-1', _score: 0.95, content: 'Machine learning' },
        { _id: 'doc-2', _score: 0.82, content: 'Vector databases' },
    ],
    namespace: 'test-ns',
    usage: { readUnits: 5 },
};
const buildMockApi = (overrides = {}) => ({
    searchDocuments: jest.fn().mockResolvedValue(mockResponse),
    ...overrides,
});
const validOptions = {
    scoreBy: [{ type: 'text', field: 'content', query: 'machine learning' }],
    topK: 5,
};
describe('searchPreviewDocuments', () => {
    describe('validation', () => {
        test('throws PineconeArgumentError when scoreBy is an empty array', async () => {
            await expect((0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', {
                ...validOptions,
                scoreBy: [],
            })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references `scoreBy` when it is empty', async () => {
            await expect((0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', {
                ...validOptions,
                scoreBy: [],
            })).rejects.toThrow('`scoreBy`');
        });
        test('throws PineconeArgumentError when scoreBy is missing', async () => {
            await expect((0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', {
                topK: 5,
            })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('throws PineconeArgumentError when topK is less than 1', async () => {
            await expect((0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', {
                ...validOptions,
                topK: 0,
            })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
        test('error message references `topK` when it is invalid', async () => {
            await expect((0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', {
                ...validOptions,
                topK: 0,
            })).rejects.toThrow('`topK`');
        });
        test('throws PineconeArgumentError when topK is missing', async () => {
            await expect((0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', {
                scoreBy: validOptions.scoreBy,
            })).rejects.toThrow(errors_1.PineconeArgumentError);
        });
    });
    describe('success', () => {
        test('returns PreviewSearchDocumentsResponse with matches, namespace, usage', async () => {
            const result = await (0, searchDocuments_1.searchPreviewDocuments)(buildMockApi(), 'test-ns', validOptions);
            expect(result.matches).toBeDefined();
            expect(result.namespace).toBe('test-ns');
            expect(result.usage).toBeDefined();
            expect(result.usage.readUnits).toBeGreaterThanOrEqual(0);
        });
        test('passes namespace to the API call', async () => {
            const api = buildMockApi();
            await (0, searchDocuments_1.searchPreviewDocuments)(api, 'my-namespace', validOptions);
            expect(api.searchDocuments).toHaveBeenCalledWith(expect.objectContaining({ namespace: 'my-namespace' }));
        });
        test('passes scoreBy and topK to the API call', async () => {
            const api = buildMockApi();
            await (0, searchDocuments_1.searchPreviewDocuments)(api, 'test-ns', validOptions);
            expect(api.searchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                searchDocumentsRequest: expect.objectContaining({
                    scoreBy: validOptions.scoreBy,
                    topK: 5,
                }),
            }));
        });
        test('passes includeFields when provided', async () => {
            const api = buildMockApi();
            await (0, searchDocuments_1.searchPreviewDocuments)(api, 'test-ns', {
                ...validOptions,
                includeFields: ['content', 'title'],
            });
            expect(api.searchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                searchDocumentsRequest: expect.objectContaining({
                    includeFields: ['content', 'title'],
                }),
            }));
        });
        test('passes filter when provided', async () => {
            const api = buildMockApi();
            const filter = { category: 'news' };
            await (0, searchDocuments_1.searchPreviewDocuments)(api, 'test-ns', {
                ...validOptions,
                filter,
            });
            expect(api.searchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                searchDocumentsRequest: expect.objectContaining({ filter }),
            }));
        });
        test('passes xPineconeApiVersion containing 2026 in every call', async () => {
            const api = buildMockApi();
            await (0, searchDocuments_1.searchPreviewDocuments)(api, 'test-ns', validOptions);
            expect(api.searchDocuments).toHaveBeenCalledWith(expect.objectContaining({
                xPineconeApiVersion: expect.stringContaining('2026'),
            }));
        });
    });
    describe('error handling', () => {
        test('converts ResponseError (404) to a typed Pinecone error via handleApiError', async () => {
            const { ResponseError } = await Promise.resolve().then(() => __importStar(require('../../../pinecone-generated-ts-fetch-alpha/db_data')));
            const api = buildMockApi({
                searchDocuments: jest.fn().mockRejectedValue(new ResponseError(new Response('{"code": 5, "message": "Index not found"}', {
                    status: 404,
                }), 'Not Found')),
            });
            await expect((0, searchDocuments_1.searchPreviewDocuments)(api, 'test-ns', validOptions)).rejects.toBeDefined();
        });
    });
});
