// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/BooleanField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BooleanFieldTypeEnum = void 0;
exports.instanceOfBooleanField = instanceOfBooleanField;
exports.BooleanFieldFromJSON = BooleanFieldFromJSON;
exports.BooleanFieldFromJSONTyped = BooleanFieldFromJSONTyped;
exports.BooleanFieldToJSON = BooleanFieldToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.BooleanFieldTypeEnum = {
    Boolean: 'boolean'
};
/**
 * Check if a given object implements the BooleanField interface.
 */
function instanceOfBooleanField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function BooleanFieldFromJSON(json) {
    return BooleanFieldFromJSONTyped(json, false);
}
function BooleanFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'filterable': !(0, runtime_1.exists)(json, 'filterable') ? undefined : json['filterable'],
    };
}
function BooleanFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'description': value.description,
        'filterable': value.filterable,
    };
}
