// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/ByocDeployment.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ByocDeploymentDeploymentTypeEnum = void 0;
exports.instanceOfByocDeployment = instanceOfByocDeployment;
exports.ByocDeploymentFromJSON = ByocDeploymentFromJSON;
exports.ByocDeploymentFromJSONTyped = ByocDeploymentFromJSONTyped;
exports.ByocDeploymentToJSON = ByocDeploymentToJSON;
/**
 * @export
 */
exports.ByocDeploymentDeploymentTypeEnum = {
    Byoc: 'byoc'
};
/**
 * Check if a given object implements the ByocDeployment interface.
 */
function instanceOfByocDeployment(value) {
    let isInstance = true;
    isInstance = isInstance && "deploymentType" in value;
    isInstance = isInstance && "environment" in value;
    return isInstance;
}
function ByocDeploymentFromJSON(json) {
    return ByocDeploymentFromJSONTyped(json, false);
}
function ByocDeploymentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deploymentType': json['deployment_type'],
        'environment': json['environment'],
    };
}
function ByocDeploymentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deployment_type': value.deploymentType,
        'environment': value.environment,
    };
}
