// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/ConfigureIndexRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfConfigureIndexRequest = instanceOfConfigureIndexRequest;
exports.ConfigureIndexRequestFromJSON = ConfigureIndexRequestFromJSON;
exports.ConfigureIndexRequestFromJSONTyped = ConfigureIndexRequestFromJSONTyped;
exports.ConfigureIndexRequestToJSON = ConfigureIndexRequestToJSON;
const runtime_1 = require("../runtime");
const PatchIndexDeploymentRequest_1 = require("./PatchIndexDeploymentRequest");
const PatchIndexSchema_1 = require("./PatchIndexSchema");
const ReadCapacity_1 = require("./ReadCapacity");
/**
 * Check if a given object implements the ConfigureIndexRequest interface.
 */
function instanceOfConfigureIndexRequest(value) {
    let isInstance = true;
    return isInstance;
}
function ConfigureIndexRequestFromJSON(json) {
    return ConfigureIndexRequestFromJSONTyped(json, false);
}
function ConfigureIndexRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deployment': !(0, runtime_1.exists)(json, 'deployment') ? undefined : (0, PatchIndexDeploymentRequest_1.PatchIndexDeploymentRequestFromJSON)(json['deployment']),
        'schema': !(0, runtime_1.exists)(json, 'schema') ? undefined : (0, PatchIndexSchema_1.PatchIndexSchemaFromJSON)(json['schema']),
        'readCapacity': !(0, runtime_1.exists)(json, 'read_capacity') ? undefined : (0, ReadCapacity_1.ReadCapacityFromJSON)(json['read_capacity']),
        'tags': !(0, runtime_1.exists)(json, 'tags') ? undefined : json['tags'],
        'deletionProtection': !(0, runtime_1.exists)(json, 'deletion_protection') ? undefined : json['deletion_protection'],
    };
}
function ConfigureIndexRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deployment': (0, PatchIndexDeploymentRequest_1.PatchIndexDeploymentRequestToJSON)(value.deployment),
        'schema': (0, PatchIndexSchema_1.PatchIndexSchemaToJSON)(value.schema),
        'read_capacity': (0, ReadCapacity_1.ReadCapacityToJSON)(value.readCapacity),
        'tags': value.tags,
        'deletion_protection': value.deletionProtection,
    };
}
