// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/CreateIndexSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateIndexSchema = instanceOfCreateIndexSchema;
exports.CreateIndexSchemaFromJSON = CreateIndexSchemaFromJSON;
exports.CreateIndexSchemaFromJSONTyped = CreateIndexSchemaFromJSONTyped;
exports.CreateIndexSchemaToJSON = CreateIndexSchemaToJSON;
const runtime_1 = require("../runtime");
const CreateIndexSchemaField_1 = require("./CreateIndexSchemaField");
/**
 * Check if a given object implements the CreateIndexSchema interface.
 */
function instanceOfCreateIndexSchema(value) {
    let isInstance = true;
    isInstance = isInstance && "fields" in value;
    return isInstance;
}
function CreateIndexSchemaFromJSON(json) {
    return CreateIndexSchemaFromJSONTyped(json, false);
}
function CreateIndexSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fields': ((0, runtime_1.mapValues)(json['fields'], CreateIndexSchemaField_1.CreateIndexSchemaFieldFromJSON)),
    };
}
function CreateIndexSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fields': ((0, runtime_1.mapValues)(value.fields, CreateIndexSchemaField_1.CreateIndexSchemaFieldToJSON)),
    };
}
