// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/CreateIndexForModelRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateIndexForModelRequest = instanceOfCreateIndexForModelRequest;
exports.CreateIndexForModelRequestFromJSON = CreateIndexForModelRequestFromJSON;
exports.CreateIndexForModelRequestFromJSONTyped = CreateIndexForModelRequestFromJSONTyped;
exports.CreateIndexForModelRequestToJSON = CreateIndexForModelRequestToJSON;
const runtime_1 = require("../runtime");
const ManagedDeployment_1 = require("./ManagedDeployment");
const ReadCapacity_1 = require("./ReadCapacity");
/**
 * Check if a given object implements the CreateIndexForModelRequest interface.
 */
function instanceOfCreateIndexForModelRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "field" in value;
    isInstance = isInstance && "model" in value;
    return isInstance;
}
function CreateIndexForModelRequestFromJSON(json) {
    return CreateIndexForModelRequestFromJSONTyped(json, false);
}
function CreateIndexForModelRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'deployment': !(0, runtime_1.exists)(json, 'deployment') ? undefined : (0, ManagedDeployment_1.ManagedDeploymentFromJSON)(json['deployment']),
        'field': json['field'],
        'model': json['model'],
        'metric': !(0, runtime_1.exists)(json, 'metric') ? undefined : json['metric'],
        'writeParameters': !(0, runtime_1.exists)(json, 'write_parameters') ? undefined : json['write_parameters'],
        'readParameters': !(0, runtime_1.exists)(json, 'read_parameters') ? undefined : json['read_parameters'],
        'readCapacity': !(0, runtime_1.exists)(json, 'read_capacity') ? undefined : (0, ReadCapacity_1.ReadCapacityFromJSON)(json['read_capacity']),
        'tags': !(0, runtime_1.exists)(json, 'tags') ? undefined : json['tags'],
        'deletionProtection': !(0, runtime_1.exists)(json, 'deletion_protection') ? undefined : json['deletion_protection'],
    };
}
function CreateIndexForModelRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'deployment': (0, ManagedDeployment_1.ManagedDeploymentToJSON)(value.deployment),
        'field': value.field,
        'model': value.model,
        'metric': value.metric,
        'write_parameters': value.writeParameters,
        'read_parameters': value.readParameters,
        'read_capacity': (0, ReadCapacity_1.ReadCapacityToJSON)(value.readCapacity),
        'tags': value.tags,
        'deletion_protection': value.deletionProtection,
    };
}
