// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/CreateIndexRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateIndexRequest = instanceOfCreateIndexRequest;
exports.CreateIndexRequestFromJSON = CreateIndexRequestFromJSON;
exports.CreateIndexRequestFromJSONTyped = CreateIndexRequestFromJSONTyped;
exports.CreateIndexRequestToJSON = CreateIndexRequestToJSON;
const runtime_1 = require("../runtime");
const CreateIndexSchema_1 = require("./CreateIndexSchema");
const IndexDeploymentRequest_1 = require("./IndexDeploymentRequest");
const ReadCapacity_1 = require("./ReadCapacity");
/**
 * Check if a given object implements the CreateIndexRequest interface.
 */
function instanceOfCreateIndexRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "schema" in value;
    return isInstance;
}
function CreateIndexRequestFromJSON(json) {
    return CreateIndexRequestFromJSONTyped(json, false);
}
function CreateIndexRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'deployment': !(0, runtime_1.exists)(json, 'deployment') ? undefined : (0, IndexDeploymentRequest_1.IndexDeploymentRequestFromJSON)(json['deployment']),
        'schema': (0, CreateIndexSchema_1.CreateIndexSchemaFromJSON)(json['schema']),
        'sourceCollection': !(0, runtime_1.exists)(json, 'source_collection') ? undefined : json['source_collection'],
        'sourceBackupId': !(0, runtime_1.exists)(json, 'source_backup_id') ? undefined : json['source_backup_id'],
        'cmekId': !(0, runtime_1.exists)(json, 'cmek_id') ? undefined : json['cmek_id'],
        'readCapacity': !(0, runtime_1.exists)(json, 'read_capacity') ? undefined : (0, ReadCapacity_1.ReadCapacityFromJSON)(json['read_capacity']),
        'tags': !(0, runtime_1.exists)(json, 'tags') ? undefined : json['tags'],
        'deletionProtection': !(0, runtime_1.exists)(json, 'deletion_protection') ? undefined : json['deletion_protection'],
    };
}
function CreateIndexRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'deployment': (0, IndexDeploymentRequest_1.IndexDeploymentRequestToJSON)(value.deployment),
        'schema': (0, CreateIndexSchema_1.CreateIndexSchemaToJSON)(value.schema),
        'source_collection': value.sourceCollection,
        'source_backup_id': value.sourceBackupId,
        'cmek_id': value.cmekId,
        'read_capacity': (0, ReadCapacity_1.ReadCapacityToJSON)(value.readCapacity),
        'tags': value.tags,
        'deletion_protection': value.deletionProtection,
    };
}
