// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/DenseVectorField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DenseVectorFieldTypeEnum = void 0;
exports.instanceOfDenseVectorField = instanceOfDenseVectorField;
exports.DenseVectorFieldFromJSON = DenseVectorFieldFromJSON;
exports.DenseVectorFieldFromJSONTyped = DenseVectorFieldFromJSONTyped;
exports.DenseVectorFieldToJSON = DenseVectorFieldToJSON;
/**
 * @export
 */
exports.DenseVectorFieldTypeEnum = {
    DenseVector: 'dense_vector',
    SparseVector: 'sparse_vector',
    SemanticText: 'semantic_text',
    String: 'string',
    StringList: 'string_list',
    Float: 'float',
    Boolean: 'boolean'
};
/**
 * Check if a given object implements the DenseVectorField interface.
 */
function instanceOfDenseVectorField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "dimension" in value;
    isInstance = isInstance && "metric" in value;
    return isInstance;
}
function DenseVectorFieldFromJSON(json) {
    return DenseVectorFieldFromJSONTyped(json, false);
}
function DenseVectorFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'dimension': json['dimension'],
        'metric': json['metric'],
    };
}
function DenseVectorFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'dimension': value.dimension,
        'metric': value.metric,
    };
}
