// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/FloatField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FloatFieldTypeEnum = void 0;
exports.instanceOfFloatField = instanceOfFloatField;
exports.FloatFieldFromJSON = FloatFieldFromJSON;
exports.FloatFieldFromJSONTyped = FloatFieldFromJSONTyped;
exports.FloatFieldToJSON = FloatFieldToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.FloatFieldTypeEnum = {
    Float: 'float'
};
/**
 * Check if a given object implements the FloatField interface.
 */
function instanceOfFloatField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function FloatFieldFromJSON(json) {
    return FloatFieldFromJSONTyped(json, false);
}
function FloatFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'filterable': !(0, runtime_1.exists)(json, 'filterable') ? undefined : json['filterable'],
    };
}
function FloatFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'description': value.description,
        'filterable': value.filterable,
    };
}
