// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/IndexSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfIndexSchema = instanceOfIndexSchema;
exports.IndexSchemaFromJSON = IndexSchemaFromJSON;
exports.IndexSchemaFromJSONTyped = IndexSchemaFromJSONTyped;
exports.IndexSchemaToJSON = IndexSchemaToJSON;
const runtime_1 = require("../runtime");
const IndexSchemaField_1 = require("./IndexSchemaField");
/**
 * Check if a given object implements the IndexSchema interface.
 */
function instanceOfIndexSchema(value) {
    let isInstance = true;
    isInstance = isInstance && "fields" in value;
    return isInstance;
}
function IndexSchemaFromJSON(json) {
    return IndexSchemaFromJSONTyped(json, false);
}
function IndexSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fields': ((0, runtime_1.mapValues)(json['fields'], IndexSchemaField_1.IndexSchemaFieldFromJSON)),
    };
}
function IndexSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fields': ((0, runtime_1.mapValues)(value.fields, IndexSchemaField_1.IndexSchemaFieldToJSON)),
    };
}
