// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/IndexModel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfIndexModel = instanceOfIndexModel;
exports.IndexModelFromJSON = IndexModelFromJSON;
exports.IndexModelFromJSONTyped = IndexModelFromJSONTyped;
exports.IndexModelToJSON = IndexModelToJSON;
const runtime_1 = require("../runtime");
const IndexDeployment_1 = require("./IndexDeployment");
const IndexModelStatus_1 = require("./IndexModelStatus");
const IndexSchema_1 = require("./IndexSchema");
const ReadCapacityResponse_1 = require("./ReadCapacityResponse");
/**
 * Check if a given object implements the IndexModel interface.
 */
function instanceOfIndexModel(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "deployment" in value;
    isInstance = isInstance && "schema" in value;
    isInstance = isInstance && "deletionProtection" in value;
    return isInstance;
}
function IndexModelFromJSON(json) {
    return IndexModelFromJSONTyped(json, false);
}
function IndexModelFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'host': json['host'],
        'privateHost': !(0, runtime_1.exists)(json, 'private_host') ? undefined : json['private_host'],
        'status': (0, IndexModelStatus_1.IndexModelStatusFromJSON)(json['status']),
        'deployment': (0, IndexDeployment_1.IndexDeploymentFromJSON)(json['deployment']),
        'readCapacity': !(0, runtime_1.exists)(json, 'read_capacity') ? undefined : (0, ReadCapacityResponse_1.ReadCapacityResponseFromJSON)(json['read_capacity']),
        'sourceCollection': !(0, runtime_1.exists)(json, 'source_collection') ? undefined : json['source_collection'],
        'sourceBackupId': !(0, runtime_1.exists)(json, 'source_backup_id') ? undefined : json['source_backup_id'],
        'cmekId': !(0, runtime_1.exists)(json, 'cmek_id') ? undefined : json['cmek_id'],
        'schema': (0, IndexSchema_1.IndexSchemaFromJSON)(json['schema']),
        'tags': !(0, runtime_1.exists)(json, 'tags') ? undefined : json['tags'],
        'deletionProtection': json['deletion_protection'],
    };
}
function IndexModelToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'host': value.host,
        'private_host': value.privateHost,
        'status': (0, IndexModelStatus_1.IndexModelStatusToJSON)(value.status),
        'deployment': (0, IndexDeployment_1.IndexDeploymentToJSON)(value.deployment),
        'read_capacity': (0, ReadCapacityResponse_1.ReadCapacityResponseToJSON)(value.readCapacity),
        'source_collection': value.sourceCollection,
        'source_backup_id': value.sourceBackupId,
        'cmek_id': value.cmekId,
        'schema': (0, IndexSchema_1.IndexSchemaToJSON)(value.schema),
        'tags': value.tags,
        'deletion_protection': value.deletionProtection,
    };
}
