// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/ManagedDeployment.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedDeploymentDeploymentTypeEnum = void 0;
exports.instanceOfManagedDeployment = instanceOfManagedDeployment;
exports.ManagedDeploymentFromJSON = ManagedDeploymentFromJSON;
exports.ManagedDeploymentFromJSONTyped = ManagedDeploymentFromJSONTyped;
exports.ManagedDeploymentToJSON = ManagedDeploymentToJSON;
/**
 * @export
 */
exports.ManagedDeploymentDeploymentTypeEnum = {
    Managed: 'managed'
};
/**
 * Check if a given object implements the ManagedDeployment interface.
 */
function instanceOfManagedDeployment(value) {
    let isInstance = true;
    isInstance = isInstance && "deploymentType" in value;
    isInstance = isInstance && "cloud" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
function ManagedDeploymentFromJSON(json) {
    return ManagedDeploymentFromJSONTyped(json, false);
}
function ManagedDeploymentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deploymentType': json['deployment_type'],
        'cloud': json['cloud'],
        'region': json['region'],
    };
}
function ManagedDeploymentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deployment_type': value.deploymentType,
        'cloud': value.cloud,
        'region': value.region,
    };
}
