// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/PatchIndexSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPatchIndexSchema = instanceOfPatchIndexSchema;
exports.PatchIndexSchemaFromJSON = PatchIndexSchemaFromJSON;
exports.PatchIndexSchemaFromJSONTyped = PatchIndexSchemaFromJSONTyped;
exports.PatchIndexSchemaToJSON = PatchIndexSchemaToJSON;
const runtime_1 = require("../runtime");
const PatchSemanticTextField_1 = require("./PatchSemanticTextField");
/**
 * Check if a given object implements the PatchIndexSchema interface.
 */
function instanceOfPatchIndexSchema(value) {
    let isInstance = true;
    isInstance = isInstance && "fields" in value;
    return isInstance;
}
function PatchIndexSchemaFromJSON(json) {
    return PatchIndexSchemaFromJSONTyped(json, false);
}
function PatchIndexSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fields': ((0, runtime_1.mapValues)(json['fields'], PatchSemanticTextField_1.PatchSemanticTextFieldFromJSON)),
    };
}
function PatchIndexSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fields': ((0, runtime_1.mapValues)(value.fields, PatchSemanticTextField_1.PatchSemanticTextFieldToJSON)),
    };
}
