// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/PatchSemanticTextField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchSemanticTextFieldTypeEnum = void 0;
exports.instanceOfPatchSemanticTextField = instanceOfPatchSemanticTextField;
exports.PatchSemanticTextFieldFromJSON = PatchSemanticTextFieldFromJSON;
exports.PatchSemanticTextFieldFromJSONTyped = PatchSemanticTextFieldFromJSONTyped;
exports.PatchSemanticTextFieldToJSON = PatchSemanticTextFieldToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.PatchSemanticTextFieldTypeEnum = {
    SemanticText: 'semantic_text'
};
/**
 * Check if a given object implements the PatchSemanticTextField interface.
 */
function instanceOfPatchSemanticTextField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function PatchSemanticTextFieldFromJSON(json) {
    return PatchSemanticTextFieldFromJSONTyped(json, false);
}
function PatchSemanticTextFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'model': !(0, runtime_1.exists)(json, 'model') ? undefined : json['model'],
        'writeParameters': !(0, runtime_1.exists)(json, 'write_parameters') ? undefined : json['write_parameters'],
        'readParameters': !(0, runtime_1.exists)(json, 'read_parameters') ? undefined : json['read_parameters'],
    };
}
function PatchSemanticTextFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'model': value.model,
        'write_parameters': value.writeParameters,
        'read_parameters': value.readParameters,
    };
}
