// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/PodDeployment.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PodDeploymentDeploymentTypeEnum = void 0;
exports.instanceOfPodDeployment = instanceOfPodDeployment;
exports.PodDeploymentFromJSON = PodDeploymentFromJSON;
exports.PodDeploymentFromJSONTyped = PodDeploymentFromJSONTyped;
exports.PodDeploymentToJSON = PodDeploymentToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.PodDeploymentDeploymentTypeEnum = {
    Pod: 'pod'
};
/**
 * Check if a given object implements the PodDeployment interface.
 */
function instanceOfPodDeployment(value) {
    let isInstance = true;
    isInstance = isInstance && "deploymentType" in value;
    isInstance = isInstance && "environment" in value;
    isInstance = isInstance && "podType" in value;
    return isInstance;
}
function PodDeploymentFromJSON(json) {
    return PodDeploymentFromJSONTyped(json, false);
}
function PodDeploymentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deploymentType': json['deployment_type'],
        'environment': json['environment'],
        'replicas': !(0, runtime_1.exists)(json, 'replicas') ? undefined : json['replicas'],
        'shards': !(0, runtime_1.exists)(json, 'shards') ? undefined : json['shards'],
        'podType': json['pod_type'],
    };
}
function PodDeploymentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deployment_type': value.deploymentType,
        'environment': value.environment,
        'replicas': value.replicas,
        'shards': value.shards,
        'pod_type': value.podType,
    };
}
