// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/ResponseStringFieldFullTextSearch.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfResponseStringFieldFullTextSearch = instanceOfResponseStringFieldFullTextSearch;
exports.ResponseStringFieldFullTextSearchFromJSON = ResponseStringFieldFullTextSearchFromJSON;
exports.ResponseStringFieldFullTextSearchFromJSONTyped = ResponseStringFieldFullTextSearchFromJSONTyped;
exports.ResponseStringFieldFullTextSearchToJSON = ResponseStringFieldFullTextSearchToJSON;
/**
 * Check if a given object implements the ResponseStringFieldFullTextSearch interface.
 */
function instanceOfResponseStringFieldFullTextSearch(value) {
    let isInstance = true;
    isInstance = isInstance && "language" in value;
    isInstance = isInstance && "stemming" in value;
    isInstance = isInstance && "stopWords" in value;
    return isInstance;
}
function ResponseStringFieldFullTextSearchFromJSON(json) {
    return ResponseStringFieldFullTextSearchFromJSONTyped(json, false);
}
function ResponseStringFieldFullTextSearchFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'language': json['language'],
        'stemming': json['stemming'],
        'stopWords': json['stop_words'],
    };
}
function ResponseStringFieldFullTextSearchToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'language': value.language,
        'stemming': value.stemming,
        'stop_words': value.stopWords,
    };
}
