// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/ResponseStringField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseStringFieldTypeEnum = void 0;
exports.instanceOfResponseStringField = instanceOfResponseStringField;
exports.ResponseStringFieldFromJSON = ResponseStringFieldFromJSON;
exports.ResponseStringFieldFromJSONTyped = ResponseStringFieldFromJSONTyped;
exports.ResponseStringFieldToJSON = ResponseStringFieldToJSON;
const runtime_1 = require("../runtime");
const ResponseStringFieldFullTextSearch_1 = require("./ResponseStringFieldFullTextSearch");
/**
 * @export
 */
exports.ResponseStringFieldTypeEnum = {
    String: 'string'
};
/**
 * Check if a given object implements the ResponseStringField interface.
 */
function instanceOfResponseStringField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function ResponseStringFieldFromJSON(json) {
    return ResponseStringFieldFromJSONTyped(json, false);
}
function ResponseStringFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'fullTextSearch': !(0, runtime_1.exists)(json, 'full_text_search') ? undefined : (0, ResponseStringFieldFullTextSearch_1.ResponseStringFieldFullTextSearchFromJSON)(json['full_text_search']),
        'filterable': !(0, runtime_1.exists)(json, 'filterable') ? undefined : json['filterable'],
    };
}
function ResponseStringFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'description': value.description,
        'full_text_search': (0, ResponseStringFieldFullTextSearch_1.ResponseStringFieldFullTextSearchToJSON)(value.fullTextSearch),
        'filterable': value.filterable,
    };
}
