// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/SemanticTextField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SemanticTextFieldTypeEnum = void 0;
exports.instanceOfSemanticTextField = instanceOfSemanticTextField;
exports.SemanticTextFieldFromJSON = SemanticTextFieldFromJSON;
exports.SemanticTextFieldFromJSONTyped = SemanticTextFieldFromJSONTyped;
exports.SemanticTextFieldToJSON = SemanticTextFieldToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.SemanticTextFieldTypeEnum = {
    SemanticText: 'semantic_text'
};
/**
 * Check if a given object implements the SemanticTextField interface.
 */
function instanceOfSemanticTextField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "model" in value;
    return isInstance;
}
function SemanticTextFieldFromJSON(json) {
    return SemanticTextFieldFromJSONTyped(json, false);
}
function SemanticTextFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'model': json['model'],
        'metric': !(0, runtime_1.exists)(json, 'metric') ? undefined : json['metric'],
        'writeParameters': !(0, runtime_1.exists)(json, 'write_parameters') ? undefined : json['write_parameters'],
        'readParameters': !(0, runtime_1.exists)(json, 'read_parameters') ? undefined : json['read_parameters'],
    };
}
function SemanticTextFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'model': value.model,
        'metric': value.metric,
        'write_parameters': value.writeParameters,
        'read_parameters': value.readParameters,
    };
}
