// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/SparseVectorField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SparseVectorFieldTypeEnum = void 0;
exports.instanceOfSparseVectorField = instanceOfSparseVectorField;
exports.SparseVectorFieldFromJSON = SparseVectorFieldFromJSON;
exports.SparseVectorFieldFromJSONTyped = SparseVectorFieldFromJSONTyped;
exports.SparseVectorFieldToJSON = SparseVectorFieldToJSON;
/**
 * @export
 */
exports.SparseVectorFieldTypeEnum = {
    SparseVector: 'sparse_vector'
};
/**
 * Check if a given object implements the SparseVectorField interface.
 */
function instanceOfSparseVectorField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function SparseVectorFieldFromJSON(json) {
    return SparseVectorFieldFromJSONTyped(json, false);
}
function SparseVectorFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
    };
}
function SparseVectorFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
    };
}
