// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/StringFieldFullTextSearch.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfStringFieldFullTextSearch = instanceOfStringFieldFullTextSearch;
exports.StringFieldFullTextSearchFromJSON = StringFieldFullTextSearchFromJSON;
exports.StringFieldFullTextSearchFromJSONTyped = StringFieldFullTextSearchFromJSONTyped;
exports.StringFieldFullTextSearchToJSON = StringFieldFullTextSearchToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the StringFieldFullTextSearch interface.
 */
function instanceOfStringFieldFullTextSearch(value) {
    let isInstance = true;
    return isInstance;
}
function StringFieldFullTextSearchFromJSON(json) {
    return StringFieldFullTextSearchFromJSONTyped(json, false);
}
function StringFieldFullTextSearchFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'language': !(0, runtime_1.exists)(json, 'language') ? undefined : json['language'],
        'stemming': !(0, runtime_1.exists)(json, 'stemming') ? undefined : json['stemming'],
        'stopWords': !(0, runtime_1.exists)(json, 'stop_words') ? undefined : json['stop_words'],
    };
}
function StringFieldFullTextSearchToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'language': value.language,
        'stemming': value.stemming,
        'stop_words': value.stopWords,
    };
}
