// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/StringListField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringListFieldTypeEnum = void 0;
exports.instanceOfStringListField = instanceOfStringListField;
exports.StringListFieldFromJSON = StringListFieldFromJSON;
exports.StringListFieldFromJSONTyped = StringListFieldFromJSONTyped;
exports.StringListFieldToJSON = StringListFieldToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.StringListFieldTypeEnum = {
    StringList: 'string_list'
};
/**
 * Check if a given object implements the StringListField interface.
 */
function instanceOfStringListField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function StringListFieldFromJSON(json) {
    return StringListFieldFromJSONTyped(json, false);
}
function StringListFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'filterable': !(0, runtime_1.exists)(json, 'filterable') ? undefined : json['filterable'],
    };
}
function StringListFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'description': value.description,
        'filterable': value.filterable,
    };
}
