// @pinecone-database/pinecone/dist/pinecone-generated-ts-fetch-alpha/db_control/models/StringField.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pinecone Control Plane API
 * Pinecone is a vector database that makes it easy to search and retrieve billions of high-dimensional vectors.
 *
 * The version of the OpenAPI document: 2026-01.alpha
 * Contact: support@pinecone.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringFieldTypeEnum = void 0;
exports.instanceOfStringField = instanceOfStringField;
exports.StringFieldFromJSON = StringFieldFromJSON;
exports.StringFieldFromJSONTyped = StringFieldFromJSONTyped;
exports.StringFieldToJSON = StringFieldToJSON;
const runtime_1 = require("../runtime");
const StringFieldFullTextSearch_1 = require("./StringFieldFullTextSearch");
/**
 * @export
 */
exports.StringFieldTypeEnum = {
    String: 'string'
};
/**
 * Check if a given object implements the StringField interface.
 */
function instanceOfStringField(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function StringFieldFromJSON(json) {
    return StringFieldFromJSONTyped(json, false);
}
function StringFieldFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'fullTextSearch': !(0, runtime_1.exists)(json, 'full_text_search') ? undefined : (0, StringFieldFullTextSearch_1.StringFieldFullTextSearchFromJSON)(json['full_text_search']),
    };
}
function StringFieldToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'description': value.description,
        'full_text_search': (0, StringFieldFullTextSearch_1.StringFieldFullTextSearchToJSON)(value.fullTextSearch),
    };
}
